// react-native/Libraries/ART/ReactNativeART.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      l = r(d[3]),
      u = r(d[4]),
      s = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/ART/ReactNativeART.js",
      f = r(d[5]),
      c = r(d[6]),
      h = r(d[7]),
      p = r(d[8]),
      v = r(d[9]),
      y = r(d[10]),
      b = r(d[11]),
      k = r(d[12]),
      _ = r(d[13]);

  function S(t, n) {
      if (null == t || null == n) return !0;
      if (t.length !== n.length) return !0;
      for (var o = 0; o < t.length; o++)
          if (t[o] !== n[o]) return !0;
      return !1
  }
  var w = k(y.UIView, {}),
      x = {
          transform: {
              diff: S
          },
          opacity: !0
      },
      C = k(x, {
          clipping: {
              diff: S
          }
      }),
      T = k(x, {
          fill: {
              diff: S
          },
          stroke: {
              diff: S
          },
          strokeWidth: !0,
          strokeCap: !0,
          strokeJoin: !0,
          strokeDash: {
              diff: S
          }
      }),
      N = k(T, {
          d: {
              diff: S
          }
      }),
      A = k(T, {
          alignment: !0,
          frame: {
              diff: function(t, n) {
                  if (t === n) return !1;
                  if (t.font !== n.font) {
                      if (null === t.font) return !0;
                      if (null === n.font) return !0;
                      if (t.font.fontFamily !== n.font.fontFamily || t.font.fontSize !== n.font.fontSize || t.font.fontWeight !== n.font.fontWeight || t.font.fontStyle !== n.font.fontStyle) return !0
                  }
                  return S(t.lines, n.lines)
              }
          },
          path: {
              diff: S
          }
      }),
      R = b('ARTSurfaceView', function() {
          return {
              validAttributes: w,
              uiViewClassName: 'ARTSurfaceView'
          }
      }),
      W = b('ARTGroup', function() {
          return {
              validAttributes: C,
              uiViewClassName: 'ARTGroup'
          }
      }),
      J = b('ARTShape', function() {
          return {
              validAttributes: N,
              uiViewClassName: 'ARTShape'
          }
      }),
      V = b('ARTText', function() {
          return {
              validAttributes: A,
              uiViewClassName: 'ARTText'
          }
      });

  function D(t) {
      return t ? 'string' == typeof t ? t : t.length ? t.join('\n') : '' : ''
  }
  var G = (function(f) {
      function c() {
          return t(this, c), o(this, l(c).apply(this, arguments))
      }
      return u(c, f), n(c, [{
          key: "getChildContext",
          value: function() {
              return {
                  isInSurface: !0
              }
          }
      }, {
          key: "render",
          value: function() {
              var t = I(this.props.height, 0),
                  n = I(this.props.width, 0);
              return p.createElement(R, {
                  style: [this.props.style, {
                      height: t,
                      width: n
                  }],
                  __source: {
                      fileName: s,
                      lineNumber: 157
                  }
              }, this.props.children)
          }
      }]), c
  })(p.Component);

  function I(t, n) {
      return null == t ? n : +t
  }
  G.childContextTypes = {
      isInSurface: v.bool
  };
  var z = new h;

  function E(t) {
      var n = null != t.scaleX ? t.scaleX : null != t.scale ? t.scale : 1,
          o = null != t.scaleY ? t.scaleY : null != t.scale ? t.scale : 1;
      return z.transformTo(1, 0, 0, 1, 0, 0).move(t.x || 0, t.y || 0).rotate(t.rotation || 0, t.originX, t.originY).scale(n, o, t.originX, t.originY), null != t.transform && z.transform(t.transform), [z.xx, z.yx, z.xy, z.yy, z.x, z.y]
  }

  function F(t) {
      return !1 === t.visible ? 0 : null == t.opacity ? 1 : +t.opacity
  }
  var M = (function(f) {
      function c() {
          return t(this, c), o(this, l(c).apply(this, arguments))
      }
      return u(c, f), n(c, [{
          key: "render",
          value: function() {
              var t = this.props;
              return _(this.context.isInSurface, 'ART: <Group /> must be a child of a <Surface />'), p.createElement(W, {
                  opacity: F(t),
                  transform: E(t),
                  __source: {
                      fileName: s,
                      lineNumber: 232
                  }
              }, this.props.children)
          }
      }]), c
  })(p.Component);
  M.contextTypes = {
      isInSurface: v.bool.isRequired
  };
  var P = (function(f) {
          function c() {
              return t(this, c), o(this, l(c).apply(this, arguments))
          }
          return u(c, f), n(c, [{
              key: "render",
              value: function() {
                  var t = this.props,
                      n = [I(t.x, 0), I(t.y, 0), I(t.width, 0), I(t.height, 0)],
                      o = k(t);
                  return delete o.x, delete o.y, p.createElement(W, {
                      clipping: n,
                      opacity: F(t),
                      transform: E(o),
                      __source: {
                          fileName: s,
                          lineNumber: 254
                      }
                  }, this.props.children)
              }
          }]), c
      })(p.Component),
      X = 0,
      Y = 1,
      O = 2,
      j = 3;

  function q(t, n, o) {
      var l = new f(t);
      n[o + 0] = l.red / 255, n[o + 1] = l.green / 255, n[o + 2] = l.blue / 255, n[o + 3] = l.alpha
  }

  function L(t, n, o) {
      var l = 0;
      if ('length' in t)
          for (; l < t.length;) q(t[l], n, o + 4 * l), l++;
      else
          for (var u in t) q(t[u], n, o + 4 * l), l++;
      return o + 4 * l
  }

  function U(t, n, o, l, u) {
      var s, f = 0;
      if ('length' in t)
          for (; f < t.length;) s = f / (t.length - 1) * l, n[o + f] = u ? 1 - s : s, f++;
      else
          for (var c in t) s = +c * l, n[o + f] = u ? 1 - s : s, f++;
      return o + f
  }

  function $(t, n, o) {
      U(t, n, L(t, n, o), 1, !1)
  }

  function B(t, n, o) {
      var l = L(t, n, o);
      U(t, n, l = U(t, n, l = L(t, n, l), .5, !1), .5, !0)
  }

  function H(t, n) {
      var o = t[0],
          l = +n.width,
          u = +n.height;
      o === Y ? (t[1] *= l, t[2] *= u, t[3] *= l, t[4] *= u) : o === O && (t[1] *= l, t[2] *= u, t[3] *= l, t[4] *= u, t[5] *= l, t[6] *= u)
  }

  function K(t, n) {
      if (null == t) return null;
      if (t._brush) return t._bb && (H(t._brush, n), t._bb = !1), t._brush;
      var o = new f(t);
      return [X, o.red / 255, o.green / 255, o.blue / 255, o.alpha]
  }

  function Q(t) {
      if (null == t) return null;
      var n = new f(t);
      return [n.red / 255, n.green / 255, n.blue / 255, n.alpha]
  }

  function Z(t) {
      switch (t) {
          case 'butt':
              return 0;
          case 'square':
              return 2;
          default:
              return 1
      }
  }

  function tt(t) {
      switch (t) {
          case 'miter':
              return 0;
          case 'bevel':
              return 2;
          default:
              return 1
      }
  }
  var et = (function(f) {
          function h() {
              return t(this, h), o(this, l(h).apply(this, arguments))
          }
          return u(h, f), n(h, [{
              key: "render",
              value: function() {
                  var t = this.props,
                      n = t.d || D(t.children),
                      o = (n instanceof c ? n : new c(n)).toJSON();
                  return p.createElement(J, {
                      fill: K(t.fill, t),
                      opacity: F(t),
                      stroke: Q(t.stroke),
                      strokeCap: Z(t.strokeCap),
                      strokeDash: t.strokeDash || null,
                      strokeJoin: tt(t.strokeJoin),
                      strokeWidth: I(t.strokeWidth, 1),
                      transform: E(t),
                      d: o,
                      __source: {
                          fileName: s,
                          lineNumber: 420
                      }
                  })
              }
          }]), h
      })(p.Component),
      nt = {},
      rt = /^[\s"']*/,
      it = /[\s"']*$/;

  function ot(t) {
      return t.split(',')[0].replace(rt, '').replace(it, '')
  }

  function lt(t) {
      if (nt.hasOwnProperty(t)) return nt[t];
      var n = /^\s*((?:(?:normal|bold|italic)\s+)*)(?:(\d+(?:\.\d+)?)[ptexm\%]*(?:\s*\/.*?)?\s+)?\s*\"?([^\"]*)/i.exec(t);
      if (!n) return null;
      var o = ot(n[3]),
          l = +n[2] || 12,
          u = /bold/.exec(n[1]),
          s = /italic/.exec(n[1]);
      return nt[t] = {
          fontFamily: o,
          fontSize: l,
          fontWeight: u ? 'bold' : 'normal',
          fontStyle: s ? 'italic' : 'normal'
      }, nt[t]
  }

  function ut(t) {
      return null == t ? null : 'string' == typeof t ? lt(t) : {
          fontFamily: ot(t.fontFamily),
          fontSize: +t.fontSize || 12,
          fontWeight: null != t.fontWeight ? t.fontWeight.toString() : '400',
          fontStyle: t.fontStyle
      }
  }
  var at = /\n/g;

  function st(t) {
      switch (t) {
          case 'right':
              return 1;
          case 'center':
              return 2;
          default:
              return 0
      }
  }
  var ft = (function(f) {
      function h() {
          return t(this, h), o(this, l(h).apply(this, arguments))
      }
      return u(h, f), n(h, [{
          key: "render",
          value: function() {
              var t, n, o = this.props,
                  l = o.path,
                  u = l ? (l instanceof c ? l : new c(l)).toJSON() : null,
                  f = (t = o.font, n = D(o.children), {
                      font: ut(t),
                      lines: n.split(at)
                  });
              return p.createElement(V, {
                  fill: K(o.fill, o),
                  opacity: F(o),
                  stroke: Q(o.stroke),
                  strokeCap: Z(o.strokeCap),
                  strokeDash: o.strokeDash || null,
                  strokeJoin: tt(o.strokeJoin),
                  strokeWidth: I(o.strokeWidth, 1),
                  transform: E(o),
                  alignment: st(o.alignment),
                  frame: f,
                  path: u,
                  __source: {
                      fileName: s,
                      lineNumber: 522
                  }
              })
          }
      }]), h
  })(p.Component);
  var ct = {
      LinearGradient: function(t, n, o, l, u) {
          var s = Y;
          if (arguments.length < 5) {
              var f = (null == n ? 270 : n) * Math.PI / 180,
                  c = Math.cos(f),
                  h = -Math.sin(f),
                  p = (Math.abs(c) + Math.abs(h)) / 2;
              n = .5 - (c *= p), l = .5 + c, o = .5 - (h *= p), u = .5 + h, this._bb = !0
          } else this._bb = !1;
          var v = [s, +n, +o, +l, +u];
          $(t, v, 5), this._brush = v
      },
      RadialGradient: function(t, n, o, l, u, s, f) {
          null == u && (u = l), null == s && (s = n), null == f && (f = o), null == n ? (n = o = l = u = s = f = .5, this._bb = !0) : this._bb = !1;
          var c = [O, +n, +o, 2 * +l, 2 * +u, +s, +f];
          B(t, c, 7), this._brush = c
      },
      Pattern: function(t, n, o, l, u) {
          this._brush = [j, t, +l || 0, +u || 0, +n, +o]
      },
      Transform: h,
      Path: c,
      Surface: G,
      Group: M,
      ClippingRectangle: P,
      Shape: et,
      Text: ft
  };
  m.exports = ct
}
