// react-native/Libraries/Components/Slider/Slider.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var n, t = r(d[0]),
      o = r(d[1]),
      l = (r(d[2]), r(d[3])),
      u = r(d[4]),
      s = (r(d[5]), r(d[6])),
      v = u.forwardRef(function(v, f) {
          var c = s.compose(n.slider, v.style),
              p = v.onValueChange,
              C = v.onSlidingComplete,
              S = o(v, ["onValueChange", "onSlidingComplete"]),
              b = p ? function(n) {
                  null != n.nativeEvent.fromUser && n.nativeEvent.fromUser && p(n.nativeEvent.value)
              } : null,
              h = b,
              E = C ? function(n) {
                  C(n.nativeEvent.value)
              } : null;
          return u.createElement(l, t({}, S, {
              ref: f,
              style: c,
              onChange: h,
              onSlidingComplete: E,
              onValueChange: b,
              enabled: !v.disabled,
              onStartShouldSetResponder: function() {
                  return !0
              },
              onResponderTerminationRequest: function() {
                  return !1
              },
              __source: {
                  fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/Slider/Slider.js",
                  lineNumber: 230
              }
          }))
      });
  v.defaultProps = {
      disabled: !1,
      value: 0,
      minimumValue: 0,
      maximumValue: 1,
      step: 0
  }, n = s.create({
      slider: {}
  }), m.exports = v
}
