// react-native-elements/src/searchbar/SearchBar-android.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[2])),
      l = t(r(d[3])),
      c = t(r(d[4])),
      u = t(r(d[5])),
      s = t(r(d[6])),
      f = t(r(d[7])),
      p = t(r(d[8])),
      y = t(r(d[9])),
      h = t(r(d[10])),
      S = n(r(d[11])),
      C = r(d[12]),
      b = r(d[13]),
      I = r(d[14]),
      w = t(r(d[15])),
      v = t(r(d[16])),
      P = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/searchbar/SearchBar-android.js";

  function T(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function O(t) {
      for (var n = 1; n < arguments.length; n++) {
          var o = null != arguments[n] ? arguments[n] : {};
          n % 2 ? T(o, !0).forEach(function(n) {
              (0, l.default)(t, n, o[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : T(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var _ = {
          type: 'material',
          size: 25,
          color: "rgba(0, 0, 0, 0.54)",
          name: 'search'
      },
      j = {
          type: 'material',
          size: 25,
          color: "rgba(0, 0, 0, 0.54)",
          name: 'arrow-back'
      },
      N = {
          type: 'material',
          size: 25,
          color: "rgba(0, 0, 0, 0.54)",
          name: 'clear'
      },
      E = (function(t) {
          function n(t) {
              var o;
              (0, u.default)(this, n), (o = (0, f.default)(this, (0, p.default)(n).call(this, t))).focus = function() {
                  o.input.focus()
              }, o.blur = function() {
                  o.input.blur()
              }, o.clear = function() {
                  o.input.clear(), o.onChangeText(''), o.props.onClear()
              }, o.cancel = function() {
                  o.blur(), o.props.onCancel()
              }, o.onFocus = function() {
                  o.props.onFocus(), o.setState({
                      hasFocus: !0,
                      isEmpty: '' === o.props.value
                  })
              }, o.onBlur = function() {
                  o.props.onBlur(), o.setState({
                      hasFocus: !1
                  })
              }, o.onChangeText = function(t) {
                  o.props.onChangeText(t), o.setState({
                      isEmpty: '' === t
                  })
              };
              var l = t.value;
              return o.state = {
                  hasFocus: !1,
                  isEmpty: !l || '' === l
              }, o
          }
          return (0, y.default)(n, t), (0, s.default)(n, [{
              key: "render",
              value: function() {
                  var t = this,
                      n = this.props,
                      l = n.clearIcon,
                      u = n.containerStyle,
                      s = n.leftIconContainerStyle,
                      f = n.rightIconContainerStyle,
                      p = n.inputContainerStyle,
                      y = n.inputStyle,
                      h = n.searchIcon,
                      b = n.cancelIcon,
                      T = n.showLoading,
                      E = n.loadingProps,
                      x = (0, c.default)(n, ["clearIcon", "containerStyle", "leftIconContainerStyle", "rightIconContainerStyle", "inputContainerStyle", "inputStyle", "searchIcon", "cancelIcon", "showLoading", "loadingProps"]),
                      B = this.state,
                      k = B.hasFocus,
                      D = B.isEmpty,
                      L = E.style,
                      V = (0, c.default)(E, ["style"]);
                  return S.default.createElement(C.View, {
                      style: C.StyleSheet.flatten([F.container, u]),
                      __source: {
                          fileName: P,
                          lineNumber: 99
                      }
                  }, S.default.createElement(w.default, (0, o.default)({
                      testID: "searchInput"
                  }, x, {
                      onFocus: this.onFocus,
                      onBlur: this.onBlur,
                      onChangeText: this.onChangeText,
                      ref: function(n) {
                          t.input = n
                      },
                      containerStyle: {
                          paddingHorizontal: 0
                      },
                      inputStyle: C.StyleSheet.flatten([F.input, y]),
                      inputContainerStyle: C.StyleSheet.flatten([F.inputContainer, p]),
                      leftIcon: k ? (0, I.renderNode)(v.default, b, O({}, j, {
                          onPress: this.cancel
                      })) : (0, I.renderNode)(v.default, h, _),
                      leftIconContainerStyle: C.StyleSheet.flatten([F.leftIconContainerStyle, s]),
                      rightIcon: S.default.createElement(C.View, {
                          style: {
                              flexDirection: 'row'
                          },
                          __source: {
                              fileName: P,
                              lineNumber: 128
                          }
                      }, T && S.default.createElement(C.ActivityIndicator, (0, o.default)({
                          key: "loading",
                          style: C.StyleSheet.flatten([{
                              marginRight: 5
                          }, L])
                      }, V, {
                          __source: {
                              fileName: P,
                              lineNumber: 130
                          }
                      })), !D && (0, I.renderNode)(v.default, l, O({}, N, {
                          key: 'cancel',
                          onPress: this.clear
                      }))),
                      rightIconContainerStyle: C.StyleSheet.flatten([F.rightIconContainerStyle, f]),
                      __source: {
                          fileName: P,
                          lineNumber: 100
                      }
                  })))
              }
          }]), n
      })(S.Component);
  E.propTypes = {
      value: h.default.string,
      clearIcon: I.nodeType,
      searchIcon: I.nodeType,
      cancelIcon: I.nodeType,
      loadingProps: h.default.object,
      showLoading: h.default.bool,
      containerStyle: b.ViewPropTypes.style,
      leftIconContainerStyle: b.ViewPropTypes.style,
      rightIconContainerStyle: b.ViewPropTypes.style,
      inputContainerStyle: b.ViewPropTypes.style,
      inputStyle: C.Text.propTypes.style,
      onClear: h.default.func,
      onCancel: h.default.func,
      onFocus: h.default.func,
      onBlur: h.default.func,
      onChangeText: h.default.func
  }, E.defaultProps = {
      value: '',
      loadingProps: {},
      showLoading: !1,
      onClear: function() {
          return null
      },
      onCancel: function() {
          return null
      },
      onFocus: function() {
          return null
      },
      onBlur: function() {
          return null
      },
      onChangeText: function() {
          return null
      },
      searchIcon: _,
      clearIcon: N,
      cancelIcon: j
  };
  var F = C.StyleSheet.create({
          container: {
              backgroundColor: 'white',
              paddingTop: 8,
              paddingBottom: 8
          },
          input: {
              marginLeft: 24,
              marginRight: 8
          },
          inputContainer: {
              borderBottomWidth: 0,
              width: '100%'
          },
          rightIconContainerStyle: {
              marginRight: 8
          },
          leftIconContainerStyle: {
              marginLeft: 8
          }
      }),
      x = E;
  e.default = x
}
