// react-native-elements/src/searchbar/SearchBar.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.SearchBar = void 0;
  var l = t(r(d[1])),
      n = t(r(d[2])),
      u = t(r(d[3])),
      c = t(r(d[4])),
      f = t(r(d[5])),
      o = t(r(d[6])),
      s = t(r(d[7])),
      h = t(r(d[8])),
      p = r(d[9]),
      b = t(r(d[10])),
      v = t(r(d[11])),
      _ = t(r(d[12])),
      y = {
          ios: b.default,
          android: v.default,
          default: _.default
      },
      w = (function(t) {
          function h() {
              var t, l;
              (0, n.default)(this, h);
              for (var u = arguments.length, o = new Array(u), s = 0; s < u; s++) o[s] = arguments[s];
              return (l = (0, c.default)(this, (t = (0, f.default)(h)).call.apply(t, [this].concat(o)))).focus = function() {
                  l.searchbar.focus()
              }, l.blur = function() {
                  l.searchbar.blur()
              }, l.clear = function() {
                  l.searchbar.clear()
              }, l.cancel = function() {
                  l.searchbar.cancel && l.searchbar.cancel()
              }, l
          }
          return (0, o.default)(h, t), (0, u.default)(h, [{
              key: "render",
              value: function() {
                  var t = this,
                      n = y[this.props.platform] || _.default;
                  return s.default.createElement(n, (0, l.default)({
                      ref: function(l) {
                          t.searchbar = l
                      }
                  }, this.props, {
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/searchbar/SearchBar.js",
                          lineNumber: 38
                      }
                  }))
              }
          }]), h
      })(s.default.Component);
  e.SearchBar = w, w.propTypes = {
      platform: h.default.oneOf(['default', 'ios', 'android'])
  }, w.defaultProps = {
      platform: 'default'
  };
  var B = (0, p.withTheme)(w, 'SearchBar');
  e.default = B
}
