// react-native-elements/src/tooltip/Triangle.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[1])),
      l = t(r(d[2])),
      n = r(d[3]),
      s = t(r(d[4])),
      u = function(t) {
          var l = t.style,
              s = t.isDown;
          return o.default.createElement(n.View, {
              style: n.StyleSheet.flatten([f.triangle, l, s ? f.down : {}]),
              __source: {
                  fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/tooltip/Triangle.js",
                  lineNumber: 8
              }
          })
      };
  u.propTypes = {
      style: s.default.style,
      isDown: l.default.bool
  };
  var f = n.StyleSheet.create({
          down: {
              transform: [{
                  rotate: '180deg'
              }]
          },
          triangle: {
              width: 0,
              height: 0,
              backgroundColor: 'transparent',
              borderStyle: 'solid',
              borderLeftWidth: 8,
              borderRightWidth: 8,
              borderBottomWidth: 15,
              borderLeftColor: 'transparent',
              borderRightColor: 'transparent',
              borderBottomColor: 'white'
          }
      }),
      b = u;
  e.default = b
}
