// expo-linear-gradient/build/LinearGradient.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[1])),
      n = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      f = t(r(d[5])),
      s = t(r(d[6])),
      c = t(r(d[7])),
      p = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      O = r(d[11]),
      h = t(r(d[12]));

  function j(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }
  var P = (function(t) {
      function o() {
          return (0, u.default)(this, o), (0, s.default)(this, (0, c.default)(o).apply(this, arguments))
      }
      return (0, p.default)(o, t), (0, f.default)(o, [{
          key: "render",
          value: function() {
              var t = this.props,
                  o = t.colors,
                  u = t.locations,
                  f = t.start,
                  s = t.end,
                  c = (0, l.default)(t, ["colors", "locations", "start", "end"]);
              return u && o.length !== u.length && (console.warn('LinearGradient colors and locations props should be arrays of the same length'), u = u.slice(0, o.length)), b.default.createElement(h.default, (0, n.default)({}, c, {
                  colors: o.map(O.processColor),
                  locations: u,
                  startPoint: v(f),
                  endPoint: v(s),
                  __source: {
                      fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/expo-linear-gradient/build/LinearGradient.js",
                      lineNumber: 12
                  }
              }))
          }
      }]), o
  })(b.default.Component);

  function v(t) {
      if (t) {
          if (!Array.isArray(t) || 2 === t.length) return Array.isArray(t) ? t : [t.x, t.y];
          console.warn('start and end props for LinearGradient must be of the format [x,y] or {x, y}')
      }
  }
  e.default = P, P.propTypes = (function(t) {
      for (var n = 1; n < arguments.length; n++) {
          var l = null != arguments[n] ? arguments[n] : {};
          n % 2 ? j(l, !0).forEach(function(n) {
              (0, o.default)(t, n, l[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : j(l).forEach(function(o) {
              Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(l, o))
          })
      }
      return t
  })({}, O.ViewPropTypes, {
      colors: y.default.arrayOf(O.ColorPropType).isRequired,
      locations: y.default.arrayOf(y.default.number),
      start: y.default.oneOfType([y.default.arrayOf(y.default.number), y.default.object]),
      end: y.default.oneOfType([y.default.arrayOf(y.default.number), y.default.object])
  })
}
