// expo-linear-gradient/build/NativeLinearGradient.android.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[1])),
      n = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      f = t(r(d[6])),
      c = t(r(d[7])),
      b = t(r(d[8])),
      p = r(d[9]),
      R = r(d[10]),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/expo-linear-gradient/build/NativeLinearGradient.android.js",
      v = (function(t) {
          function R() {
              return (0, l.default)(this, R), (0, s.default)(this, (0, f.default)(R).apply(this, arguments))
          }
          return (0, c.default)(R, t), (0, u.default)(R, [{
              key: "render",
              value: function() {
                  var t = this.props,
                      l = t.colors,
                      u = t.locations,
                      s = t.startPoint,
                      f = t.endPoint,
                      c = t.children,
                      R = t.style,
                      v = (0, n.default)(t, ["colors", "locations", "startPoint", "endPoint", "children", "style"]),
                      _ = p.StyleSheet.flatten(R) || {},
                      P = _.borderRadius || 0,
                      L = [_.borderTopLeftRadius || P, _.borderTopLeftRadius || P, _.borderTopRightRadius || P, _.borderTopRightRadius || P, _.borderBottomRightRadius || P, _.borderBottomRightRadius || P, _.borderBottomLeftRadius || P, _.borderBottomLeftRadius || P];
                  return b.default.createElement(p.View, (0, o.default)({}, v, {
                      style: R,
                      __source: {
                          fileName: h,
                          lineNumber: 23
                      }
                  }), b.default.createElement(y, {
                      style: {
                          position: 'absolute',
                          top: 0,
                          left: 0,
                          bottom: 0,
                          right: 0
                      },
                      colors: l,
                      startPoint: s,
                      endPoint: f,
                      locations: u,
                      borderRadii: L,
                      __source: {
                          fileName: h,
                          lineNumber: 24
                      }
                  }), c)
              }
          }]), R
      })(b.default.Component);
  e.default = v;
  var y = (0, R.requireNativeViewManager)('ExpoLinearGradient')
}
