// react-native/Libraries/Components/SegmentedControlIOS/SegmentedControlIOS.android.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      l = r(d[4]),
      u = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/SegmentedControlIOS/SegmentedControlIOS.android.js",
      c = r(d[5]),
      f = r(d[6]),
      p = r(d[7]),
      h = r(d[8]),
      y = (function(f) {
          function y() {
              return t(this, y), o(this, s(y).apply(this, arguments))
          }
          return l(y, f), n(y, [{
              key: "render",
              value: function() {
                  return c.createElement(h, {
                      style: [b.dummy, this.props.style],
                      __source: {
                          fileName: u,
                          lineNumber: 20
                      }
                  }, c.createElement(p, {
                      style: b.text,
                      __source: {
                          fileName: u,
                          lineNumber: 21
                      }
                  }, "SegmentedControlIOS is not supported on this platform!"))
              }
          }]), y
      })(c.Component),
      b = f.create({
          dummy: {
              width: 120,
              height: 50,
              backgroundColor: '#ffbcbc',
              borderWidth: 1,
              borderColor: 'red',
              alignItems: 'center',
              justifyContent: 'center'
          },
          text: {
              color: '#333333',
              margin: 5,
              fontSize: 10
          }
      });
  m.exports = y
}
