// react-native-ratings/src/components/Star.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      s = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var l = t(r(d[2])),
      n = t(r(d[3])),
      o = t(r(d[4])),
      u = t(r(d[5])),
      c = t(r(d[6])),
      f = s(r(d[7])),
      p = r(d[8]),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-ratings/src/components/Star.js",
      v = r(d[9]),
      y = r(d[10]),
      _ = (function(t) {
          function s() {
              var t;
              return (0, l.default)(this, s), (t = (0, o.default)(this, (0, u.default)(s).call(this))).springValue = new p.Animated.Value(1), t.state = {
                  selected: !1
              }, t
          }
          return (0, c.default)(s, t), (0, n.default)(s, [{
              key: "spring",
              value: function() {
                  var t = this.props,
                      s = t.position,
                      l = t.starSelectedInPosition;
                  this.springValue.setValue(1.2), p.Animated.spring(this.springValue, {
                      toValue: 1,
                      friction: 2,
                      tension: 1
                  }).start(), this.setState({
                      selected: !this.state.selected
                  }), l(s)
              }
          }, {
              key: "render",
              value: function() {
                  var t = this.props,
                      s = t.fill,
                      l = t.size,
                      n = t.selectedColor,
                      o = t.isDisabled,
                      u = t.starStyle,
                      c = s && null === n ? y : v;
                  return f.default.createElement(p.TouchableOpacity, {
                      activeOpacity: 1,
                      onPress: this.spring.bind(this),
                      disabled: o,
                      __source: {
                          fileName: h,
                          lineNumber: 45
                      }
                  }, f.default.createElement(p.Animated.Image, {
                      source: c,
                      style: [S.starStyle, {
                          tintColor: s && n ? n : void 0,
                          width: l || 40,
                          height: l || 40,
                          transform: [{
                              scale: this.springValue
                          }]
                      }, u],
                      __source: {
                          fileName: h,
                          lineNumber: 46
                      }
                  }))
              }
          }]), s
      })(f.PureComponent);
  e.default = _, _.defaultProps = {
      selectedColor: '#f1c40f'
  };
  var S = p.StyleSheet.create({
      starStyle: {
          margin: 3
      }
  })
}
