// react-native/Libraries/Image/Image.android.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var n = r(d[0]),
      t = r(d[1]),
      o = r(d[2]),
      c = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Image/Image.android.js";

  function u(n, t) {
      var o = Object.keys(n);
      if (Object.getOwnPropertySymbols) {
          var c = Object.getOwnPropertySymbols(n);
          t && (c = c.filter(function(t) {
              return Object.getOwnPropertyDescriptor(n, t).enumerable
          })), o.push.apply(o, c)
      }
      return o
  }
  var s = r(d[3]),
      l = r(d[4]),
      f = r(d[5]),
      h = r(d[6]),
      p = r(d[7]),
      b = r(d[8]),
      y = r(d[9]),
      w = (r(d[10]), r(d[11])),
      v = r(d[12]),
      O = r(d[13]),
      S = r(d[14]),
      I = r(d[15]),
      E = p.ImageLoader,
      j = r(d[16]),
      P = 1;
  var L = (function(n) {
      for (var t = 1; t < arguments.length; t++) {
          var c = null != arguments[t] ? arguments[t] : {};
          t % 2 ? u(c, !0).forEach(function(t) {
              o(n, t, c[t])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(n, Object.getOwnPropertyDescriptors(c)) : u(c).forEach(function(t) {
              Object.defineProperty(n, t, Object.getOwnPropertyDescriptor(c, t))
          })
      }
      return n
  })({}, f, {
      style: l(s),
      source: b.oneOfType([b.shape({
          uri: b.string,
          headers: b.objectOf(b.string)
      }), b.number, b.arrayOf(b.shape({
          uri: b.string,
          width: b.number,
          height: b.number,
          headers: b.objectOf(b.string)
      }))]),
      blurRadius: b.number,
      defaultSource: b.number,
      loadingIndicatorSource: b.oneOfType([b.shape({
          uri: b.string
      }), b.number]),
      progressiveRenderingEnabled: b.bool,
      fadeDuration: b.number,
      onLoadStart: b.func,
      onError: b.func,
      onLoad: b.func,
      onLoadEnd: b.func,
      testID: b.string,
      resizeMethod: b.oneOf(['auto', 'resize', 'scale']),
      resizeMode: b.oneOf(['cover', 'contain', 'stretch', 'repeat', 'center'])
  });
  var _ = function(t, o) {
      var u, s, l, f, p = I(t.source),
          b = I(t.defaultSource),
          w = I(t.loadingIndicatorSource);
      if (p && '' === p.uri && console.warn('source.uri should not be an empty string'), t.src && console.warn('The <Image> component requires a `source` property rather than `src`.'), t.children) throw new Error('The <Image> component cannot contain children. If you want to render content on top of the image, consider using the <ImageBackground> component or absolute positioning.');
      if (t.defaultSource && t.loadingIndicatorSource) throw new Error('The <Image> component cannot have defaultSource and loadingIndicatorSource at the same time. Please use either defaultSource or loadingIndicatorSource.');
      if (!p || p.uri || Array.isArray(p) || (p = null), null != (null == (u = p) ? void 0 : u.uri)) {
          var E = p,
              P = E.width,
              L = E.height;
          l = O([{
              width: P,
              height: L
          }, N.base, t.style]), f = [{
              uri: p.uri
          }]
      } else l = O([N.base, t.style]), f = p;
      var _ = t.onLoadStart,
          D = t.onLoad,
          z = t.onLoadEnd,
          T = t.onError,
          q = S(t, {
              style: l,
              shouldNotifyLoadEvents: !!(_ || D || z || T),
              src: f,
              headers: null == (s = p) ? void 0 : s.headers,
              defaultSrc: b ? b.uri : null,
              loadingIndicatorSrc: w ? w.uri : null,
              ref: o
          });
      return y.createElement(v.Consumer, {
          __source: {
              fileName: c,
              lineNumber: 249
          }
      }, function(t) {
          return t ? y.createElement(j, n({}, q, {
              __source: {
                  fileName: c,
                  lineNumber: 252
              }
          })) : y.createElement(h, n({}, q, {
              __source: {
                  fileName: c,
                  lineNumber: 254
              }
          }))
      })
  };
  (_ = y.forwardRef(_)).displayName = 'Image', _.getSize = function(n, t, o) {
      return E.getSize(n).then(function(n) {
          t(n.width, n.height)
      }).catch(o || function() {
          console.warn('Failed to get size for image: ' + n)
      })
  }, _.prefetch = function(n, t) {
      var o = P++;
      return t && t(o), E.prefetchImage(n, o)
  }, _.abortPrefetch = function(n) {
      E.abortRequest(n)
  }, _.queryCache = function(n) {
      return t.async(function(o) {
          for (;;) switch (o.prev = o.next) {
              case 0:
                  return o.next = 2, t.awrap(E.queryCache(n));
              case 2:
                  return o.abrupt("return", o.sent);
              case 3:
              case "end":
                  return o.stop()
          }
      })
  }, _.resolveAssetSource = I, _.propTypes = L;
  var N = w.create({
      base: {
          overflow: 'hidden'
      }
  });
  m.exports = _
}
