// react-native/Libraries/Image/ImageBackground.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      s = r(d[2]),
      l = r(d[3]),
      o = r(d[4]),
      c = r(d[5]),
      u = r(d[6]),
      f = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Image/ImageBackground.js",
      h = r(d[7]),
      v = r(d[8]),
      y = r(d[9]),
      _ = r(d[10]),
      p = r(d[11]),
      w = (function(w) {
          function R() {
              var t, n;
              s(this, R);
              for (var l = arguments.length, u = new Array(l), f = 0; f < l; f++) u[f] = arguments[f];
              return (n = o(this, (t = c(R)).call.apply(t, [this].concat(u))))._viewRef = null, n._captureRef = function(t) {
                  n._viewRef = t
              }, n
          }
          return u(R, w), l(R, [{
              key: "setNativeProps",
              value: function(t) {
                  var n = this._viewRef;
                  n && (p(n), n.setNativeProps(t))
              }
          }, {
              key: "render",
              value: function() {
                  var s = this.props,
                      l = s.children,
                      o = s.style,
                      c = s.imageStyle,
                      u = s.imageRef,
                      p = n(s, ["children", "style", "imageStyle", "imageRef"]);
                  return v.createElement(_, {
                      accessibilityIgnoresInvertColors: !0,
                      style: o,
                      ref: this._captureRef,
                      __source: {
                          fileName: f,
                          lineNumber: 63
                      }
                  }, v.createElement(h, t({}, p, {
                      style: [y.absoluteFill, {
                          width: o.width,
                          height: o.height
                      }, c],
                      ref: u,
                      __source: {
                          fileName: f,
                          lineNumber: 67
                      }
                  })), l)
              }
          }]), R
      })(v.Component);
  m.exports = w
}
