// react-native/Libraries/ReactNative/AppContainer.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      l = r(d[4]),
      u = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/ReactNative/AppContainer.js",
      p = (r(d[5]), r(d[6])),
      c = (r(d[7]), r(d[8])),
      h = (r(d[9]), r(d[10])),
      f = r(d[11]),
      v = (function(p) {
          function h() {
              var n, l;
              t(this, h);
              for (var u = arguments.length, p = new Array(u), c = 0; c < u; c++) p[c] = arguments[c];
              return (l = o(this, (n = s(h)).call.apply(n, [this].concat(p)))).state = {
                  inspector: null,
                  mainKey: 1
              }, l._subscription = null, l
          }
          return l(h, p), n(h, [{
              key: "getChildContext",
              value: function() {
                  return {
                      rootTag: this.props.rootTag
                  }
              }
          }, {
              key: "componentDidMount",
              value: function() {}
          }, {
              key: "componentWillUnmount",
              value: function() {
                  null != this._subscription && this._subscription.remove()
              }
          }, {
              key: "render",
              value: function() {
                  var t = this,
                      n = c.createElement(f, {
                          collapsable: !this.state.inspector,
                          key: this.state.mainKey,
                          pointerEvents: "box-none",
                          style: y.appContainer,
                          ref: function(n) {
                              t._mainRef = n
                          },
                          __source: {
                              fileName: u,
                              lineNumber: 98
                          }
                      }, this.props.children),
                      o = this.props.WrapperComponent;
                  return null != o && (n = c.createElement(o, {
                      __source: {
                          fileName: u,
                          lineNumber: 112
                      }
                  }, n)), c.createElement(f, {
                      style: y.appContainer,
                      pointerEvents: "box-none",
                      __source: {
                          fileName: u,
                          lineNumber: 115
                      }
                  }, n, null, this.state.inspector)
              }
          }]), h
      })(c.Component);
  v.childContextTypes = {
      rootTag: p.number
  };
  var y = h.create({
      appContainer: {
          flex: 1
      }
  });
  m.exports = v
}
