// react-native-elements/src/text/Text.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.TextElement = void 0;
  var l = t(r(d[1])),
      o = t(r(d[2])),
      n = t(r(d[3])),
      y = t(r(d[4])),
      h = t(r(d[5])),
      f = r(d[6]),
      s = r(d[7]),
      p = t(r(d[8]));

  function u(t, l) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var n = Object.getOwnPropertySymbols(t);
          l && (n = n.filter(function(l) {
              return Object.getOwnPropertyDescriptor(t, l).enumerable
          })), o.push.apply(o, n)
      }
      return o
  }

  function c(t) {
      for (var o = 1; o < arguments.length; o++) {
          var n = null != arguments[o] ? arguments[o] : {};
          o % 2 ? u(n, !0).forEach(function(o) {
              (0, l.default)(t, o, n[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : u(n).forEach(function(l) {
              Object.defineProperty(t, l, Object.getOwnPropertyDescriptor(n, l))
          })
      }
      return t
  }
  var S = function(t) {
      var l = t.style,
          h = t.children,
          s = t.h1,
          u = t.h2,
          c = t.h3,
          S = t.h4,
          T = t.h1Style,
          O = t.h2Style,
          x = t.h3Style,
          v = t.h4Style,
          P = (0, n.default)(t, ["style", "children", "h1", "h2", "h3", "h4", "h1Style", "h2Style", "h3Style", "h4Style"]);
      return y.default.createElement(f.Text, (0, o.default)({
          style: f.StyleSheet.flatten([b.text, l && l, s && b.bold, u && b.bold, c && b.bold, S && b.bold, s && f.StyleSheet.flatten([{
              fontSize: (0, p.default)(40)
          }, T]), u && f.StyleSheet.flatten([{
              fontSize: (0, p.default)(34)
          }, O]), c && f.StyleSheet.flatten([{
              fontSize: (0, p.default)(28)
          }, x]), S && f.StyleSheet.flatten([{
              fontSize: (0, p.default)(22)
          }, v])])
      }, P, {
          __source: {
              fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/text/Text.js",
              lineNumber: 24
          }
      }), h)
  };
  e.TextElement = S, S.propTypes = {
      style: f.Text.propTypes.style,
      h1: h.default.bool,
      h2: h.default.bool,
      h3: h.default.bool,
      h4: h.default.bool,
      h1Style: f.Text.propTypes.style,
      h2Style: f.Text.propTypes.style,
      h3Style: f.Text.propTypes.style,
      h4Style: f.Text.propTypes.style,
      children: h.default.node
  }, S.defaultProps = {
      h1: !1,
      h2: !1,
      h3: !1,
      h4: !1,
      style: {},
      h1Style: {},
      h2Style: {},
      h3Style: {},
      h4Style: {},
      children: ''
  };
  var b = f.StyleSheet.create({
          text: c({}, f.Platform.select({
              android: c({}, s.fonts.android.regular)
          })),
          bold: c({}, f.Platform.select({
              android: c({}, s.fonts.android.bold)
          }))
      }),
      T = (0, s.withTheme)(S, 'Text');
  e.default = T
}
