// react-native-gesture-handler/GestureComponents.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = t(r(d[1])),
      l = t(r(d[2])),
      o = t(r(d[3])),
      u = t(r(d[4])),
      s = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-gesture-handler/GestureComponents.js",
      f = {};

  function c(t, n) {
      var l = f[t.displayName];
      return l || (f[t.displayName] = (0, u.default)(t, n))
  }
  m.exports = {
      get ScrollView() {
          return c(o.default.ScrollView, {
              disallowInterruption: !0
          })
      },
      get Switch() {
          return c(o.default.Switch, {
              shouldCancelWhenOutside: !1,
              shouldActivateOnStart: !0,
              disallowInterruption: !0
          })
      },
      get TextInput() {
          return c(o.default.TextInput)
      },
      get ToolbarAndroid() {
          return c(o.default.ToolbarAndroid)
      },
      get DrawerLayoutAndroid() {
          var t = c(o.default.DrawerLayoutAndroid, {
              disallowInterruption: !0
          });
          return t.positions = o.default.DrawerLayoutAndroid.positions, t
      },
      get FlatList() {
          if (!f.FlatList) {
              var t = this.ScrollView;
              f.FlatList = l.default.forwardRef(function(u, f) {
                  return l.default.createElement(o.default.FlatList, (0, n.default)({
                      ref: f
                  }, u, {
                      renderScrollComponent: function(o) {
                          return l.default.createElement(t, (0, n.default)({}, o, {
                              __source: {
                                  fileName: s,
                                  lineNumber: 53
                              }
                          }))
                      },
                      __source: {
                          fileName: s,
                          lineNumber: 50
                      }
                  }))
              })
          }
          return f.FlatList
      }
  }
}
