// react-native-gesture-handler/gestureHandlerRootHOC.android.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = function(t) {
      var s = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : void 0,
          y = (function(y) {
              function O() {
                  return (0, o.default)(this, O), (0, l.default)(this, (0, c.default)(O).apply(this, arguments))
              }
              return (0, f.default)(O, y), (0, u.default)(O, [{
                  key: "render",
                  value: function() {
                      return p.default.createElement(j, {
                          style: [w.container, s],
                          __source: {
                              fileName: b,
                              lineNumber: 28
                          }
                      }, p.default.createElement(t, (0, n.default)({}, this.props, {
                          __source: {
                              fileName: b,
                              lineNumber: 29
                          }
                      })))
                  }
              }]), O
          })(p.default.Component);
      return (0, O.default)(y, t), y
  };
  var n = t(r(d[1])),
      o = t(r(d[2])),
      u = t(r(d[3])),
      l = t(r(d[4])),
      c = t(r(d[5])),
      f = t(r(d[6])),
      s = t(r(d[7])),
      p = t(r(d[8])),
      y = r(d[9]),
      O = t(r(d[10])),
      b = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-gesture-handler/gestureHandlerRootHOC.android.js";

  function v(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var u = Object.getOwnPropertySymbols(t);
          n && (u = u.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, u)
      }
      return o
  }
  var h = {
          name: 'GestureHandlerRootView',
          propTypes: (function(t) {
              for (var n = 1; n < arguments.length; n++) {
                  var o = null != arguments[n] ? arguments[n] : {};
                  n % 2 ? v(o, !0).forEach(function(n) {
                      (0, s.default)(t, n, o[n])
                  }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : v(o).forEach(function(n) {
                      Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
                  })
              }
              return t
          })({}, y.ViewPropTypes)
      },
      j = (0, y.requireNativeComponent)('GestureHandlerRootView', h);
  var w = y.StyleSheet.create({
      container: {
          flex: 1
      }
  })
}
