// react-native-gesture-handler/GestureButtons.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), Object.defineProperty(e, "PureNativeButton", {
      enumerable: !0,
      get: function() {
          return _.default
      }
  }), e.BorderlessButton = e.RectButton = e.BaseButton = e.RawButton = void 0;
  var n = t(r(d[1])),
      o = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      p = t(r(d[6])),
      c = t(r(d[7])),
      f = t(r(d[8])),
      v = t(r(d[9])),
      h = t(r(d[10])),
      y = r(d[11]),
      C = t(r(d[12])),
      _ = t(r(d[13])),
      A = t(r(d[14])),
      S = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-gesture-handler/GestureButtons.js";

  function b(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function O(t) {
      for (var o = 1; o < arguments.length; o++) {
          var l = null != arguments[o] ? arguments[o] : {};
          o % 2 ? b(l, !0).forEach(function(o) {
              (0, n.default)(t, o, l[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : b(l).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(l, n))
          })
      }
      return t
  }
  var E = (0, C.default)(_.default, {
      shouldCancelWhenOutside: !1,
      shouldActivateOnStart: !1
  });
  e.RawButton = E;
  var P = (function(t) {
      function n(t) {
          var o;
          return (0, u.default)(this, n), (o = (0, p.default)(this, (0, c.default)(n).call(this, t)))._handleEvent = function(t) {
              var n = t.nativeEvent,
                  l = n.state,
                  u = n.oldState,
                  s = n.pointerInside && l === A.default.ACTIVE;
              s !== o._lastActive && o.props.onActiveStateChange && o.props.onActiveStateChange(s), u === A.default.ACTIVE && l !== A.default.CANCELLED && o._lastActive && o.props.onPress && o.props.onPress(s), o._lastActive = s
          }, o._onHandlerStateChange = function(t) {
              o.props.onHandlerStateChange && o.props.onHandlerStateChange(t), o._handleEvent(t)
          }, o._onGestureEvent = function(t) {
              o.props.onGestureEvent && o.props.onGestureEvent(t), o._handleEvent(t)
          }, o._lastActive = !1, o
      }
      return (0, f.default)(n, t), (0, s.default)(n, [{
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.style,
                  u = t.rippleColor,
                  s = (0, l.default)(t, ["style", "rippleColor"]);
              return h.default.createElement(E, (0, o.default)({
                  style: [{
                      overflow: 'hidden'
                  }, n],
                  rippleColor: (0, y.processColor)(u)
              }, s, {
                  onGestureEvent: this._onGestureEvent,
                  onHandlerStateChange: this._onHandlerStateChange,
                  __source: {
                      fileName: S,
                      lineNumber: 66
                  }
              }))
          }
      }]), n
  })(h.default.Component);
  e.BaseButton = P, P.propTypes = O({}, E.propTypes, {
      onPress: v.default.func,
      onActiveStateChange: v.default.func
  });
  var w = y.Animated.createAnimatedComponent(P),
      B = y.StyleSheet.create({
          underlay: {
              position: 'absolute',
              left: 0,
              right: 0,
              bottom: 0,
              top: 0
          }
      }),
      j = (function(t) {
          function n(t) {
              var o;
              return (0, u.default)(this, n), (o = (0, p.default)(this, (0, c.default)(n).call(this, t)))._onActiveStateChange = function(t) {
                  'android' !== y.Platform.OS && o._opacity.setValue(t ? o.props.activeOpacity : 0), o.props.onActiveStateChange && o.props.onActiveStateChange(t)
              }, o._opacity = new y.Animated.Value(0), o
          }
          return (0, f.default)(n, t), (0, s.default)(n, [{
              key: "render",
              value: function() {
                  var t = this.props,
                      n = t.children,
                      u = (0, l.default)(t, ["children"]);
                  return h.default.createElement(P, (0, o.default)({}, u, {
                      onActiveStateChange: this._onActiveStateChange,
                      __source: {
                          fileName: S,
                          lineNumber: 114
                      }
                  }), h.default.createElement(y.Animated.View, {
                      style: [B.underlay, {
                          opacity: this._opacity
                      }, {
                          backgroundColor: this.props.underlayColor
                      }],
                      __source: {
                          fileName: S,
                          lineNumber: 115
                      }
                  }), n)
              }
          }]), n
      })(h.default.Component);
  e.RectButton = j, j.propTypes = P.propTypes, j.defaultProps = {
      activeOpacity: .105,
      underlayColor: 'black'
  };
  var N = (function(t) {
      function n(t) {
          var o;
          return (0, u.default)(this, n), (o = (0, p.default)(this, (0, c.default)(n).call(this, t)))._onActiveStateChange = function(t) {
              'android' !== y.Platform.OS && o._opacity.setValue(t ? o.props.activeOpacity : 1), o.props.onActiveStateChange && o.props.onActiveStateChange(t)
          }, o._opacity = new y.Animated.Value(1), o
      }
      return (0, f.default)(n, t), (0, s.default)(n, [{
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.children,
                  u = t.style,
                  s = (0, l.default)(t, ["children", "style"]);
              return h.default.createElement(w, (0, o.default)({}, s, {
                  onActiveStateChange: this._onActiveStateChange,
                  style: [u, 'ios' === y.Platform.OS && {
                      opacity: this._opacity
                  }],
                  __source: {
                      fileName: S,
                      lineNumber: 156
                  }
              }), n)
          }
      }]), n
  })(h.default.Component);
  e.BorderlessButton = N, N.propTypes = O({}, P.propTypes, {
      borderless: v.default.bool
  }), N.defaultProps = {
      activeOpacity: .3,
      borderless: !0
  }
}
