// expo/build/launch/RootErrorBoundary.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[1])),
      o = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      c = t(r(d[6])),
      f = t(r(d[7])),
      p = r(d[8]),
      h = (r(d[9]), "/Users/linyiting/Desktop/Arrow/arrow/node_modules/expo/build/launch/RootErrorBoundary.js"),
      _ = p.NativeModules.ExponentAppLoadingManager;

  function y() {
      return c.default.async(function(t) {
          for (;;) switch (t.prev = t.next) {
              case 0:
                  if (!_ || !_.finishedAsync) {
                      t.next = 4;
                      break
                  }
                  return t.next = 3, c.default.awrap(_.finishedAsync());
              case 3:
                  return t.abrupt("return", t.sent);
              case 4:
              case "end":
                  return t.stop()
          }
      })
  }
  var v = (function(t) {
      function c() {
          var t;
          return (0, n.default)(this, c), (t = (0, l.default)(this, (0, u.default)(c).apply(this, arguments))).state = {
              error: null
          }, t._appLoadingIsMounted = !1, t._subscribeToGlobalErrors = function() {
              t._appLoadingIsMounted = !0;
              var n = g.ErrorUtils;
              if (n) {
                  var o = n.getGlobalHandler();
                  n.setGlobalHandler(function(n, l) {
                      t._appLoadingIsMounted && (y(), l && t.setState({
                          error: n
                      })), o(n, l)
                  })
              }
          }, t._unsubscribeFromGlobalErrors = function() {
              t._appLoadingIsMounted = !1
          }, t
      }
      return (0, s.default)(c, t), (0, o.default)(c, [{
          key: "componentWillMount",
          value: function() {}
      }, {
          key: "componentDidCatch",
          value: function(t) {
              this._appLoadingIsMounted && (y(), this.setState({
                  error: t
              })), console.error(t)
          }
      }, {
          key: "render",
          value: function() {
              return this.state.error ? f.default.createElement(p.View, {
                  style: b.container,
                  __source: {
                      fileName: h,
                      lineNumber: 60
                  }
              }, f.default.createElement(p.Text, {
                  style: b.warningIcon,
                  __source: {
                      fileName: h,
                      lineNumber: 61
                  }
              }, "\u26a0\ufe0f"), f.default.createElement(p.Text, {
                  style: [b.paragraph, {
                      color: '#000'
                  }],
                  __source: {
                      fileName: h,
                      lineNumber: 62
                  }
              }, "A fatal error was encountered while rendering the root component."), f.default.createElement(p.Text, {
                  style: b.paragraph,
                  __source: {
                      fileName: h,
                      lineNumber: 65
                  }
              }, "Review your application logs for more information, and reload the app when the issue is resolved. In production, your app would have crashed.")) : this.props.children
          }
      }]), c
  })(f.default.Component);
  e.default = v;
  var b = p.StyleSheet.create({
      container: {
          flex: 1,
          alignItems: 'center',
          justifyContent: 'center'
      },
      paragraph: {
          marginBottom: 10,
          textAlign: 'center',
          marginHorizontal: 30,
          maxWidth: 350,
          fontSize: 15,
          color: '#888'
      },
      warningIcon: {
          fontSize: 40,
          marginBottom: 20
      }
  })
}
