// react-native/Libraries/Lists/VirtualizedSectionList.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      l = r(d[4]),
      c = r(d[5]),
      p = r(d[6]),
      u = r(d[7]),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Lists/VirtualizedSectionList.js";

  function f(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }

  function S(t) {
      for (var o = 1; o < arguments.length; o++) {
          var s = null != arguments[o] ? arguments[o] : {};
          o % 2 ? f(s, !0).forEach(function(o) {
              n(t, o, s[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(s)) : f(s).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(s, n))
          })
      }
      return t
  }
  r(d[8]);
  var v = r(d[9]),
      _ = r(d[10]),
      y = r(d[11]),
      I = r(d[12]),
      b = (function(n) {
          function f(t, n) {
              var p;
              return o(this, f), (p = s(this, l(f).call(this, t, n)))._keyExtractor = function(t, n) {
                  var o = p._subExtractor(n);
                  return o && o.key || String(n)
              }, p._convertViewable = function(t) {
                  I(null != t.index, 'Received a broken ViewToken');
                  var n = p._subExtractor(t.index);
                  if (!n) return null;
                  var o = n.section.keyExtractor || p.props.keyExtractor;
                  return S({}, t, {
                      index: n.index,
                      key: o(t.item, n.index),
                      section: n.section
                  })
              }, p._onViewableItemsChanged = function(t) {
                  var n = t.viewableItems,
                      o = t.changed;
                  p.props.onViewableItemsChanged && p.props.onViewableItemsChanged({
                      viewableItems: n.map(p._convertViewable, c(p)).filter(Boolean),
                      changed: o.map(p._convertViewable, c(p)).filter(Boolean)
                  })
              }, p._renderItem = function(t) {
                  var n = t.item,
                      o = t.index,
                      s = p._subExtractor(o);
                  if (!s) return null;
                  var l = s.index;
                  if (null == l) {
                      var c = s.section;
                      if (!0 === s.header) {
                          var u = p.props.renderSectionHeader;
                          return u ? u({
                              section: c
                          }) : null
                      }
                      var f = p.props.renderSectionFooter;
                      return f ? f({
                          section: c
                      }) : null
                  }
                  var S = s.section.renderItem || p.props.renderItem,
                      _ = p._getSeparatorComponent(o, s);
                  return I(S, 'no renderItem!'), v.createElement(P, {
                      SeparatorComponent: _,
                      LeadingSeparatorComponent: 0 === l ? p.props.SectionSeparatorComponent : void 0,
                      cellKey: s.key,
                      index: l,
                      item: n,
                      leadingItem: s.leadingItem,
                      leadingSection: s.leadingSection,
                      onUpdateSeparator: p._onUpdateSeparator,
                      prevCellKey: (p._subExtractor(o - 1) || {}).key,
                      ref: function(t) {
                          p._cellRefs[s.key] = t
                      },
                      renderItem: S,
                      section: s.section,
                      trailingItem: s.trailingItem,
                      trailingSection: s.trailingSection,
                      __source: {
                          fileName: h,
                          lineNumber: 320
                      }
                  })
              }, p._onUpdateSeparator = function(t, n) {
                  var o = p._cellRefs[t];
                  o && o.updateSeparatorProps(n)
              }, p._cellRefs = {}, p._captureRef = function(t) {
                  p._listRef = t
              }, p.state = p._computeState(t), p
          }
          return u(f, n), p(f, [{
              key: "scrollToLocation",
              value: function(t) {
                  for (var n = t.itemIndex + 1, o = 0; o < t.sectionIndex; o++) n += this.props.sections[o].data.length + 2;
                  var s = S({}, t, {
                      index: n
                  });
                  this._listRef.scrollToIndex(s)
              }
          }, {
              key: "getListRef",
              value: function() {
                  return this._listRef
              }
          }]), p(f, [{
              key: "UNSAFE_componentWillReceiveProps",
              value: function(t) {
                  this.setState(this._computeState(t))
              }
          }, {
              key: "_computeState",
              value: function(t) {
                  var n = t.ListHeaderComponent ? 1 : 0,
                      o = [],
                      s = t.sections.reduce(function(t, s) {
                          return o.push(t + n), t + s.data.length + 2
                      }, 0);
                  return {
                      childProps: S({}, t, {
                          renderItem: this._renderItem,
                          ItemSeparatorComponent: void 0,
                          data: t.sections,
                          getItemCount: function() {
                              return s
                          },
                          getItem: k,
                          keyExtractor: this._keyExtractor,
                          onViewableItemsChanged: t.onViewableItemsChanged ? this._onViewableItemsChanged : void 0,
                          stickyHeaderIndices: t.stickySectionHeadersEnabled ? o : void 0
                      })
                  }
              }
          }, {
              key: "render",
              value: function() {
                  return v.createElement(y, t({}, this.state.childProps, {
                      ref: this._captureRef,
                      __source: {
                          fileName: h,
                          lineNumber: 202
                      }
                  }))
              }
          }, {
              key: "_subExtractor",
              value: function(t) {
                  for (var n = t, o = this.props.sections, s = 0; s < o.length; s++) {
                      var l = o[s],
                          c = l.key || String(s);
                      if ((n -= 1) >= l.data.length + 1) n -= l.data.length + 1;
                      else return -1 === n ? {
                          section: l,
                          key: c + ':header',
                          index: null,
                          header: !0,
                          trailingSection: o[s + 1]
                      } : n === l.data.length ? {
                          section: l,
                          key: c + ':footer',
                          index: null,
                          header: !1,
                          trailingSection: o[s + 1]
                      } : {
                          section: l,
                          key: c + ':' + (l.keyExtractor || this.props.keyExtractor)(l.data[n], n),
                          index: n,
                          leadingItem: l.data[n - 1],
                          leadingSection: o[s - 1],
                          trailingItem: l.data.length > n + 1 ? l.data[n + 1] : void 0,
                          trailingSection: o[s + 1]
                      }
                  }
              }
          }, {
              key: "_getSeparatorComponent",
              value: function(t, n) {
                  if (!(n = n || this._subExtractor(t))) return null;
                  var o = n.section.ItemSeparatorComponent || this.props.ItemSeparatorComponent,
                      s = this.props.SectionSeparatorComponent,
                      l = t === this.state.childProps.getItemCount() - 1,
                      c = n.index === n.section.data.length - 1;
                  return s && c ? s : !o || c || l ? null : o
              }
          }]), f
      })(v.PureComponent);
  b.defaultProps = S({}, y.defaultProps, {
      data: []
  });
  var P = (function(n) {
      function c() {
          var t, n;
          o(this, c);
          for (var p = arguments.length, u = new Array(p), h = 0; h < p; h++) u[h] = arguments[h];
          return (n = s(this, (t = l(c)).call.apply(t, [this].concat(u)))).state = {
              separatorProps: {
                  highlighted: !1,
                  leadingItem: n.props.item,
                  leadingSection: n.props.leadingSection,
                  section: n.props.section,
                  trailingItem: n.props.trailingItem,
                  trailingSection: n.props.trailingSection
              },
              leadingSeparatorProps: {
                  highlighted: !1,
                  leadingItem: n.props.leadingItem,
                  leadingSection: n.props.leadingSection,
                  section: n.props.section,
                  trailingItem: n.props.item,
                  trailingSection: n.props.trailingSection
              }
          }, n._separators = {
              highlight: function() {
                  ['leading', 'trailing'].forEach(function(t) {
                      return n._separators.updateProps(t, {
                          highlighted: !0
                      })
                  })
              },
              unhighlight: function() {
                  ['leading', 'trailing'].forEach(function(t) {
                      return n._separators.updateProps(t, {
                          highlighted: !1
                      })
                  })
              },
              updateProps: function(t, o) {
                  var s = n.props,
                      l = s.LeadingSeparatorComponent,
                      c = s.cellKey,
                      p = s.prevCellKey;
                  'leading' === t && null != l ? n.setState(function(t) {
                      return {
                          leadingSeparatorProps: S({}, t.leadingSeparatorProps, {}, o)
                      }
                  }) : n.props.onUpdateSeparator('leading' === t && p || c, o)
              }
          }, n
      }
      return u(c, n), p(c, [{
          key: "updateSeparatorProps",
          value: function(t) {
              this.setState(function(n) {
                  return {
                      separatorProps: S({}, n.separatorProps, {}, t)
                  }
              })
          }
      }, {
          key: "render",
          value: function() {
              var n = this.props,
                  o = n.LeadingSeparatorComponent,
                  s = n.SeparatorComponent,
                  l = n.item,
                  c = n.index,
                  p = n.section,
                  u = this.props.renderItem({
                      item: l,
                      index: c,
                      section: p,
                      separators: this._separators
                  }),
                  f = o && v.createElement(o, t({}, this.state.leadingSeparatorProps, {
                      __source: {
                          fileName: h,
                          lineNumber: 506
                      }
                  })),
                  S = s && v.createElement(s, t({}, this.state.separatorProps, {
                      __source: {
                          fileName: h,
                          lineNumber: 509
                      }
                  }));
              return f || S ? v.createElement(_, {
                  __source: {
                      fileName: h,
                      lineNumber: 515
                  }
              }, f, u, S) : u
          }
      }], [{
          key: "getDerivedStateFromProps",
          value: function(t, n) {
              return {
                  separatorProps: S({}, n.separatorProps, {
                      leadingItem: t.item,
                      leadingSection: t.leadingSection,
                      section: t.section,
                      trailingItem: t.trailingItem,
                      trailingSection: t.trailingSection
                  }),
                  leadingSeparatorProps: S({}, n.leadingSeparatorProps, {
                      leadingItem: t.leadingItem,
                      leadingSection: t.leadingSection,
                      section: t.section,
                      trailingItem: t.item,
                      trailingSection: t.trailingSection
                  })
              }
          }
      }]), c
  })(v.Component);

  function k(t, n) {
      if (!t) return null;
      for (var o = n - 1, s = 0; s < t.length; s++) {
          if (-1 === o || o === t[s].data.length) return t[s];
          if (o < t[s].data.length) return t[s].data[o];
          o -= t[s].data.length + 2
      }
      return null
  }
  m.exports = b
}
