// react-native-elements/src/overlay/Overlay.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Overlay = void 0;
  var o = t(r(d[1])),
      l = t(r(d[2])),
      n = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      c = r(d[6]),
      f = r(d[7]),
      y = r(d[8]),
      b = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/overlay/Overlay.js";

  function h(t, o) {
      var l = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var n = Object.getOwnPropertySymbols(t);
          o && (n = n.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), l.push.apply(l, n)
      }
      return l
  }
  var p = function(t) {
      var o = t.children,
          s = t.isVisible,
          f = t.containerStyle,
          y = t.overlayStyle,
          h = t.windowBackgroundColor,
          p = t.overlayBackgroundColor,
          v = t.onBackdropPress,
          O = t.borderRadius,
          k = t.width,
          S = t.height,
          _ = t.fullScreen,
          P = (0, n.default)(t, ["children", "isVisible", "containerStyle", "overlayStyle", "windowBackgroundColor", "overlayBackgroundColor", "onBackdropPress", "borderRadius", "width", "height", "fullScreen"]);
      return u.default.createElement(c.Modal, (0, l.default)({
          visible: s,
          onRequestClose: v,
          transparent: !0
      }, P, {
          __source: {
              fileName: b,
              lineNumber: 31
          }
      }), u.default.createElement(c.TouchableWithoutFeedback, {
          onPress: v,
          testID: "RNE__Overlay__backdrop",
          __source: {
              fileName: b,
              lineNumber: 37
          }
      }, u.default.createElement(c.View, {
          testID: "overlayContainer",
          style: c.StyleSheet.flatten([w.backdrop, {
              backgroundColor: h
          }, f]),
          __source: {
              fileName: b,
              lineNumber: 41
          }
      })), u.default.createElement(c.View, {
          style: w.container,
          pointerEvents: "box-none",
          __source: {
              fileName: b,
              lineNumber: 51
          }
      }, u.default.createElement(c.View, {
          style: c.StyleSheet.flatten([w.overlay, {
              borderRadius: O,
              backgroundColor: p,
              width: k,
              height: S
          }, _ && w.fullscreen, y]),
          __source: {
              fileName: b,
              lineNumber: 52
          }
      }, o)))
  };
  e.Overlay = p, p.propTypes = {
      children: s.default.element.isRequired,
      isVisible: s.default.bool.isRequired,
      containerStyle: f.ViewPropTypes.style,
      overlayStyle: f.ViewPropTypes.style,
      windowBackgroundColor: s.default.string,
      overlayBackgroundColor: s.default.string,
      onBackdropPress: s.default.func,
      borderRadius: s.default.number,
      width: s.default.oneOfType([s.default.string, s.default.number]),
      height: s.default.oneOfType([s.default.string, s.default.number]),
      fullScreen: s.default.bool
  }, p.defaultProps = {
      borderRadius: 3,
      fullScreen: !1,
      windowBackgroundColor: 'rgba(0, 0, 0, .4)',
      overlayBackgroundColor: 'white',
      width: y.ScreenWidth - 80,
      height: y.ScreenHeight - 180,
      onBackdropPress: function() {
          return null
      }
  };
  var w = c.StyleSheet.create({
          backdrop: {
              position: 'absolute',
              top: 0,
              left: 0,
              width: '100%',
              height: '100%',
              justifyContent: 'center',
              alignItems: 'center'
          },
          container: {
              flex: 1,
              alignItems: 'center',
              justifyContent: 'center'
          },
          fullscreen: {
              width: '100%',
              height: '100%'
          },
          overlay: (function(t) {
              for (var l = 1; l < arguments.length; l++) {
                  var n = null != arguments[l] ? arguments[l] : {};
                  l % 2 ? h(n, !0).forEach(function(l) {
                      (0, o.default)(t, l, n[l])
                  }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : h(n).forEach(function(o) {
                      Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
                  })
              }
              return t
          })({
              borderRadius: 5,
              padding: 10
          }, c.Platform.select({
              android: {
                  elevation: 2
              },
              default: {
                  shadowColor: 'rgba(0, 0, 0, .3)',
                  shadowOffset: {
                      width: 0,
                      height: 1
                  },
                  shadowRadius: 4
              }
          }))
      }),
      v = (0, f.withTheme)(p, 'Overlay');
  e.default = v
}
