// @skele/components/src/viewport/withPlaceholder/index.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var l = t(r(d[1])),
      n = t(r(d[2])),
      o = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      p = t(r(d[6])),
      f = t(r(d[7])),
      c = t(r(d[8])),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/@skele/components/src/viewport/withPlaceholder/index.js";
  e.default = function(t, _) {
      var w, N;
      return N = w = (function(c) {
          function w(t, l) {
              return (0, n.default)(this, w), (0, u.default)(this, (0, s.default)(w).call(this, t, l))
          }
          return (0, p.default)(w, c), (0, o.default)(w, [{
              key: "componentDidUpdate",
              value: function() {
                  this.props.inViewport && !this._hasEnteredViewport && (this._hasEnteredViewport = !0)
              }
          }, {
              key: "render",
              value: function() {
                  return this.props.inViewport ? f.default.createElement(t, (0, l.default)({}, this.props, {
                      __source: {
                          fileName: h,
                          lineNumber: 20
                      }
                  })) : this.props.retainOnceInViewport && this._hasEnteredViewport ? f.default.createElement(t, (0, l.default)({}, this.props, {
                      __source: {
                          fileName: h,
                          lineNumber: 24
                      }
                  })) : this.props.placeholder ? f.default.createElement(this.props.placeholder, {
                      __source: {
                          fileName: h,
                          lineNumber: 28
                      }
                  }) : _ ? f.default.createElement(_, {
                      __source: {
                          fileName: h,
                          lineNumber: 30
                      }
                  }) : null
              }
          }]), w
      })(f.default.Component), w.propTypes = {
          inViewport: c.default.bool.isRequired,
          placeholder: c.default.func,
          retainOnceInViewport: c.default.bool
      }, w.displayName = "WithPlaceholder(" + (t.displayName || t.name || 'Component') + ")", N
  }
}
