// react-native/Libraries/Experimental/SwipeableRow/SwipeableRow.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var n = r(d[0]),
      t = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      u = r(d[4]),
      l = r(d[5]),
      p = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Experimental/SwipeableRow/SwipeableRow.js",
      h = r(d[6]),
      _ = r(d[7]),
      c = r(d[8]),
      v = r(d[9]),
      f = r(d[10]),
      w = r(d[11]),
      R = _.isRTL,
      S = 0,
      P = 300,
      y = function() {},
      T = (function(_) {
          function f() {
              var n, o;
              t(this, f);
              for (var l = arguments.length, p = new Array(l), _ = 0; _ < l; _++) p[_] = arguments[_];
              return (o = s(this, (n = u(f)).call.apply(n, [this].concat(p))))._handleMoveShouldSetPanResponderCapture = function(n, t) {
                  return t.dy < 10 && o._isValidSwipe(t)
              }, o._handlePanResponderGrant = function(n, t) {}, o._handlePanResponderMove = function(n, t) {
                  o._isSwipingExcessivelyRightFromClosedPosition(t) || (o.props.onSwipeStart && o.props.onSwipeStart(), o._isSwipingRightFromClosed(t) ? o._swipeSlowSpeed(t) : o._swipeFullSpeed(t))
              }, o._onPanResponderTerminationRequest = function(n, t) {
                  return !1
              }, o._handlePanResponderEnd = function(n, t) {
                  var s = R ? -t.dx : t.dx;
                  o._isSwipingRightFromClosed(t) ? (o.props.onOpen && o.props.onOpen(), o._animateBounceBack(P)) : o._shouldAnimateRemainder(t) ? s < 0 ? (o.props.onOpen && o.props.onOpen(), o._animateToOpenPositionWith(t.vx, s)) : (o.props.onClose && o.props.onClose(), o._animateToClosedPosition()) : o._previousLeft === S ? o._animateToClosedPosition() : o._animateToOpenPosition(), o.props.onSwipeEnd && o.props.onSwipeEnd()
              }, o._panResponder = c.create({
                  onMoveShouldSetPanResponderCapture: o._handleMoveShouldSetPanResponderCapture,
                  onPanResponderGrant: o._handlePanResponderGrant,
                  onPanResponderMove: o._handlePanResponderMove,
                  onPanResponderRelease: o._handlePanResponderEnd,
                  onPanResponderTerminationRequest: o._onPanResponderTerminationRequest,
                  onPanResponderTerminate: o._handlePanResponderEnd,
                  onShouldBlockNativeResponder: function(n, t) {
                      return !1
                  }
              }), o._previousLeft = S, o._timeoutID = null, o.state = {
                  currentLeft: new h.Value(o._previousLeft),
                  isSwipeableViewRendered: !1,
                  rowHeight: null
              }, o._onSwipeableViewLayout = function(n) {
                  o.setState({
                      isSwipeableViewRendered: !0,
                      rowHeight: n.nativeEvent.layout.height
                  })
              }, o._animateToClosedPositionDuringBounce = function() {
                  o._animateToClosedPosition(P)
              }, o
          }
          return l(f, _), o(f, [{
              key: "componentDidMount",
              value: function() {
                  var n = this;
                  this.props.shouldBounceOnMount && (this._timeoutID = setTimeout(function() {
                      n._animateBounceBack(400)
                  }, 700))
              }
          }, {
              key: "UNSAFE_componentWillReceiveProps",
              value: function(n) {
                  var t, o, s = null != (t = this.props.isOpen) && t,
                      u = null != (o = n.isOpen) && o;
                  s && !u && this._animateToClosedPosition()
              }
          }, {
              key: "componentWillUnmount",
              value: function() {
                  null != this._timeoutID && clearTimeout(this._timeoutID)
              }
          }, {
              key: "render",
              value: function() {
                  var t;
                  this.state.isSwipeableViewRendered && this.state.rowHeight && (t = v.createElement(w, {
                      style: [C.slideOutContainer, {
                          height: this.state.rowHeight
                      }],
                      __source: {
                          fileName: p,
                          lineNumber: 214
                      }
                  }, this.props.slideoutView));
                  var o = v.createElement(h.View, {
                      onLayout: this._onSwipeableViewLayout,
                      style: {
                          transform: [{
                              translateX: this.state.currentLeft
                          }]
                      },
                      __source: {
                          fileName: p,
                          lineNumber: 223
                      }
                  }, this.props.children);
                  return v.createElement(w, n({}, this._panResponder.panHandlers, {
                      __source: {
                          fileName: p,
                          lineNumber: 231
                      }
                  }), t, o)
              }
          }, {
              key: "close",
              value: function() {
                  this.props.onClose && this.props.onClose(), this._animateToClosedPosition()
              }
          }, {
              key: "_isSwipingRightFromClosed",
              value: function(n) {
                  var t = R ? -n.dx : n.dx;
                  return this._previousLeft === S && t > 0
              }
          }, {
              key: "_swipeFullSpeed",
              value: function(n) {
                  this.state.currentLeft.setValue(this._previousLeft + n.dx)
              }
          }, {
              key: "_swipeSlowSpeed",
              value: function(n) {
                  this.state.currentLeft.setValue(this._previousLeft + n.dx / 4)
              }
          }, {
              key: "_isSwipingExcessivelyRightFromClosedPosition",
              value: function(n) {
                  var t = R ? -n.dx : n.dx;
                  return this._isSwipingRightFromClosed(n) && t > 120
              }
          }, {
              key: "_animateTo",
              value: function(n) {
                  var t = this,
                      o = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : 300,
                      s = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : y;
                  h.timing(this.state.currentLeft, {
                      duration: o,
                      toValue: n,
                      useNativeDriver: !0
                  }).start(function() {
                      t._previousLeft = n, s()
                  })
              }
          }, {
              key: "_animateToOpenPosition",
              value: function() {
                  var n, t = null != (n = this.props.maxSwipeDistance) ? n : 0,
                      o = R ? -t : t;
                  this._animateTo(-o)
              }
          }, {
              key: "_animateToOpenPositionWith",
              value: function(n, t) {
                  var o;
                  n = n > .3 ? n : .3;
                  var s = null != (o = this.props.maxSwipeDistance) ? o : 0,
                      u = Math.abs((s - Math.abs(t)) / n),
                      l = R ? -s : s;
                  this._animateTo(-l, u)
              }
          }, {
              key: "_animateToClosedPosition",
              value: function() {
                  var n = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : 300;
                  this._animateTo(S, n)
              }
          }, {
              key: "_animateBounceBack",
              value: function(n) {
                  var t = R ? -30 : 30;
                  this._animateTo(-t, n, this._animateToClosedPositionDuringBounce)
              }
          }, {
              key: "_isValidSwipe",
              value: function(n) {
                  var t;
                  return !(null != (t = this.props.preventSwipeRight) && t && this._previousLeft === S && n.dx > 0) && Math.abs(n.dx) > 10
              }
          }, {
              key: "_shouldAnimateRemainder",
              value: function(n) {
                  var t, o = null != (t = this.props.swipeThreshold) ? t : 30;
                  return Math.abs(n.dx) > o || n.vx > .3
              }
          }]), f
      })(v.Component),
      C = f.create({
          slideOutContainer: {
              bottom: 0,
              left: 0,
              position: 'absolute',
              right: 0,
              top: 0
          }
      });
  m.exports = T
}
