// react-native/Libraries/Experimental/SwipeableRow/SwipeableFlatList.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      l = r(d[4]),
      u = r(d[5]),
      c = r(d[6]),
      p = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Experimental/SwipeableRow/SwipeableFlatList.js";

  function f(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }
  var w = r(d[7]),
      _ = r(d[8]),
      S = r(d[9]),
      b = (function(t) {
          function f(t, n) {
              var s;
              return o(this, f), (s = l(this, u(f).call(this, t, n)))._flatListRef = null, s._shouldBounceFirstRowOnMount = !1, s._onScroll = function(t) {
                  s.state.openRowKey && s.setState({
                      openRowKey: null
                  }), s.props.onScroll && s.props.onScroll(t)
              }, s._renderItem = function(t) {
                  var n = s.props.renderQuickActions(t),
                      o = s.props.keyExtractor(t.item, t.index);
                  if (!n) return s.props.renderItem(t);
                  var l = !1;
                  return s._shouldBounceFirstRowOnMount && (s._shouldBounceFirstRowOnMount = !1, l = !0), w.createElement(_, {
                      slideoutView: n,
                      isOpen: o === s.state.openRowKey,
                      maxSwipeDistance: s._getMaxSwipeDistance(t),
                      onOpen: function() {
                          return s._onOpen(o)
                      },
                      onClose: function() {
                          return s._onClose(o)
                      },
                      shouldBounceOnMount: l,
                      onSwipeEnd: s._setListViewScrollable,
                      onSwipeStart: s._setListViewNotScrollable,
                      __source: {
                          fileName: p,
                          lineNumber: 122
                      }
                  }, s.props.renderItem(t))
              }, s._setListViewScrollable = function() {
                  s._setListViewScrollableTo(!0)
              }, s._setListViewNotScrollable = function() {
                  s._setListViewScrollableTo(!1)
              }, s.state = {
                  openRowKey: null
              }, s._shouldBounceFirstRowOnMount = s.props.bounceFirstRowOnMount, s
          }
          return c(f, t), s(f, [{
              key: "render",
              value: function() {
                  var t = this;
                  return w.createElement(S, n({}, this.props, {
                      ref: function(n) {
                          t._flatListRef = n
                      },
                      onScroll: this._onScroll,
                      renderItem: this._renderItem,
                      extraData: this.state,
                      __source: {
                          fileName: p,
                          lineNumber: 83
                      }
                  }))
              }
          }, {
              key: "_getMaxSwipeDistance",
              value: function(t) {
                  return 'function' == typeof this.props.maxSwipeDistance ? this.props.maxSwipeDistance(t) : this.props.maxSwipeDistance
              }
          }, {
              key: "_setListViewScrollableTo",
              value: function(t) {
                  this._flatListRef && this._flatListRef.setNativeProps({
                      scrollEnabled: t
                  })
              }
          }, {
              key: "_onOpen",
              value: function(t) {
                  this.setState({
                      openRowKey: t
                  })
              }
          }, {
              key: "_onClose",
              value: function(t) {
                  this.setState({
                      openRowKey: null
                  })
              }
          }]), f
      })(w.Component);
  b.defaultProps = (function(n) {
      for (var o = 1; o < arguments.length; o++) {
          var s = null != arguments[o] ? arguments[o] : {};
          o % 2 ? f(s, !0).forEach(function(o) {
              t(n, o, s[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(n, Object.getOwnPropertyDescriptors(s)) : f(s).forEach(function(t) {
              Object.defineProperty(n, t, Object.getOwnPropertyDescriptor(s, t))
          })
      }
      return n
  })({}, S.defaultProps, {
      bounceFirstRowOnMount: !0,
      renderQuickActions: function() {
          return null
      }
  }), m.exports = b
}
