// react-native-elements/src/config/ThemeProvider.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.ThemeConsumer = e.default = e.ThemeContext = void 0;
  var u = t(r(d[1])),
      n = t(r(d[2])),
      l = t(r(d[3])),
      o = t(r(d[4])),
      h = t(r(d[5])),
      f = t(r(d[6])),
      s = t(r(d[7])),
      c = t(r(d[8])),
      T = t(r(d[9])),
      p = f.default.createContext({
          theme: {
              colors: T.default
          }
      });
  e.ThemeContext = p;
  var v = (function(t) {
      function s(t) {
          var n;
          return (0, u.default)(this, s), (n = (0, l.default)(this, (0, o.default)(s).call(this, t))).updateTheme = function(t) {
              n.setState(function(u) {
                  var n = u.theme;
                  return {
                      theme: (0, c.default)(n, t)
                  }
              })
          }, n.replaceTheme = function(t) {
              n.setState(function() {
                  return {
                      theme: (0, c.default)(n.defaultTheme, t)
                  }
              })
          }, n.getTheme = function() {
              return n.state.theme
          }, n.defaultTheme = (0, c.default)({
              colors: T.default
          }, t.theme), n.state = {
              theme: n.defaultTheme
          }, n
      }
      return (0, h.default)(s, t), (0, n.default)(s, [{
          key: "render",
          value: function() {
              return f.default.createElement(p.Provider, {
                  value: {
                      theme: this.state.theme,
                      updateTheme: this.updateTheme,
                      replaceTheme: this.replaceTheme
                  },
                  __source: {
                      fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/config/ThemeProvider.js",
                      lineNumber: 44
                  }
              }, this.props.children)
          }
      }]), s
  })(f.default.Component);
  e.default = v, v.propTypes = {
      theme: s.default.object,
      children: s.default.node.isRequired
  }, v.defaultProps = {
      theme: {}
  };
  var C = p.Consumer;
  e.ThemeConsumer = C
}
