// react-native-elements/src/config/withTheme.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[1])),
      u = t(r(d[2])),
      l = t(r(d[3])),
      f = t(r(d[4])),
      o = t(r(d[5])),
      c = t(r(d[6])),
      p = t(r(d[7])),
      s = t(r(d[8])),
      h = t(r(d[9])),
      y = t(r(d[10])),
      b = t(r(d[11])),
      _ = r(d[12]),
      v = t(r(d[13])),
      O = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/config/withTheme.js";

  function N(t, n) {
      var u = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), u.push.apply(u, l)
      }
      return u
  }

  function w(t) {
      for (var n = 1; n < arguments.length; n++) {
          var l = null != arguments[n] ? arguments[n] : {};
          n % 2 ? N(l, !0).forEach(function(n) {
              (0, u.default)(t, n, l[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : N(l).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(l, n))
          })
      }
      return t
  }
  var j = function(t) {
          return Boolean(t.prototype && t.prototype.isReactComponent)
      },
      P = function(t, u) {
          var N = (function(b) {
                  function N() {
                      return (0, f.default)(this, N), (0, c.default)(this, (0, p.default)(N).apply(this, arguments))
                  }
                  return (0, s.default)(N, b), (0, o.default)(N, [{
                      key: "render",
                      value: function() {
                          var f = this.props,
                              o = f.forwardedRef,
                              c = f.children,
                              p = (0, l.default)(f, ["forwardedRef", "children"]);
                          return h.default.createElement(_.ThemeConsumer, {
                              __source: {
                                  fileName: O,
                                  lineNumber: 17
                              }
                          }, function(l) {
                              if (!l) {
                                  var f = w({}, p, {
                                      theme: v.default,
                                      children: c
                                  });
                                  return j(t) ? h.default.createElement(t, (0, n.default)({
                                      ref: o
                                  }, f, {
                                      __source: {
                                          fileName: O,
                                          lineNumber: 24
                                      }
                                  })) : h.default.createElement(t, (0, n.default)({}, f, {
                                      __source: {
                                          fileName: O,
                                          lineNumber: 26
                                      }
                                  }))
                              }
                              var s = l.theme,
                                  b = w({
                                      theme: s,
                                      updateTheme: l.updateTheme,
                                      replaceTheme: l.replaceTheme
                                  }, (0, y.default)(u && s[u] || {}, p), {
                                      children: c
                                  });
                              return j(t) ? h.default.createElement(t, (0, n.default)({
                                  ref: o
                              }, b, {
                                  __source: {
                                      fileName: O,
                                      lineNumber: 40
                                  }
                              })) : h.default.createElement(t, (0, n.default)({}, b, {
                                  __source: {
                                      fileName: O,
                                      lineNumber: 42
                                  }
                              }))
                          })
                      }
                  }]), N
              })(h.default.Component),
              P = u ? "Themed." + u : "Themed." + (t.displayName || t.name || 'Component');
          if (j(t)) {
              var E = function(t, u) {
                  return h.default.createElement(N, (0, n.default)({}, t, {
                      forwardedRef: u,
                      __source: {
                          fileName: O,
                          lineNumber: 57
                      }
                  }))
              };
              return E.displayName = P, (0, b.default)(h.default.forwardRef(E), t)
          }
          return N.displayName = P, N
      };
  e.default = P
}
