// react-native-elements/src/image/Image.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Image = void 0;
  var n = t(r(d[1])),
      l = t(r(d[2])),
      o = t(r(d[3])),
      c = t(r(d[4])),
      s = t(r(d[5])),
      u = t(r(d[6])),
      p = t(r(d[7])),
      f = t(r(d[8])),
      y = t(r(d[9])),
      h = t(r(d[10])),
      b = r(d[11]),
      O = r(d[12]),
      v = r(d[13]),
      _ = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/image/Image.js";

  function w(t, n) {
      var l = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var o = Object.getOwnPropertySymbols(t);
          n && (o = o.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), l.push.apply(l, o)
      }
      return l
  }

  function I(t) {
      for (var l = 1; l < arguments.length; l++) {
          var o = null != arguments[l] ? arguments[l] : {};
          l % 2 ? w(o, !0).forEach(function(l) {
              (0, n.default)(t, l, o[l])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : w(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var S = (function(t) {
      function n() {
          var t, l;
          (0, c.default)(this, n);
          for (var o = arguments.length, s = new Array(o), f = 0; f < o; f++) s[f] = arguments[f];
          return (l = (0, u.default)(this, (t = (0, p.default)(n)).call.apply(t, [this].concat(s)))).state = {
              placeholderOpacity: new b.Animated.Value(1)
          }, l.onLoad = function() {
              var t = 200 * Math.random();
              setTimeout(function() {
                  b.Animated.timing(l.state.placeholderOpacity, {
                      toValue: 0,
                      duration: 350,
                      useNativeDriver: 'android' !== b.Platform.OS
                  }).start()
              }, 'android' === b.Platform.OS ? 0 : Math.floor(100 + t))
          }, l
      }
      return (0, f.default)(n, t), (0, s.default)(n, [{
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.placeholderStyle,
                  c = t.PlaceholderContent,
                  s = t.containerStyle,
                  u = t.style,
                  p = t.ImageComponent,
                  f = t.children,
                  h = (0, o.default)(t, ["placeholderStyle", "PlaceholderContent", "containerStyle", "style", "ImageComponent", "children"]),
                  O = Boolean(h.source);
              return y.default.createElement(b.View, {
                  accessibilityIgnoresInvertColors: !0,
                  style: b.StyleSheet.flatten([P.container, s]),
                  __source: {
                      fileName: _,
                      lineNumber: 48
                  }
              }, y.default.createElement(p, (0, l.default)({}, h, {
                  onLoad: this.onLoad,
                  style: [b.StyleSheet.absoluteFill, {
                      width: u.width,
                      height: u.height
                  }],
                  testID: "RNE__Image",
                  __source: {
                      fileName: _,
                      lineNumber: 52
                  }
              })), y.default.createElement(b.Animated.View, {
                  pointerEvents: O ? 'none' : 'auto',
                  accessibilityElementsHidden: O,
                  importantForAccessibility: O ? 'no-hide-descendants' : 'yes',
                  style: [P.placeholderContainer, {
                      opacity: O ? this.state.placeholderOpacity : 1
                  }],
                  __source: {
                      fileName: _,
                      lineNumber: 65
                  }
              }, y.default.createElement(b.View, {
                  testID: "RNE__Image__placeholder",
                  style: b.StyleSheet.flatten([u, P.placeholder, n]),
                  __source: {
                      fileName: _,
                      lineNumber: 76
                  }
              }, c)), y.default.createElement(b.View, {
                  style: u,
                  __source: {
                      fileName: _,
                      lineNumber: 88
                  }
              }, f))
          }
      }]), n
  })(y.default.Component);
  e.Image = S;
  var P = {
      container: {
          backgroundColor: 'transparent',
          position: 'relative'
      },
      placeholderContainer: I({}, b.StyleSheet.absoluteFillObject),
      placeholder: {
          backgroundColor: '#bdbdbd',
          alignItems: 'center',
          justifyContent: 'center'
      }
  };
  S.propTypes = I({}, b.Image.propTypes, {
      ImageComponent: h.default.oneOfType([h.default.func, h.default.object]),
      PlaceholderContent: O.nodeType,
      containerStyle: v.ViewPropTypes.style,
      placeholderStyle: b.Image.propTypes.style
  }), S.defaultProps = {
      ImageComponent: b.Image,
      style: {}
  };
  var j = (0, v.withTheme)(S, 'Image');
  e.default = j
}
