// react-native/Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var o = r(d[0]),
      n = r(d[1]),
      t = r(d[2]),
      c = r(d[3]),
      s = r(d[4]),
      p = r(d[5]),
      l = r(d[6]),
      f = r(d[7]),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/ToolbarAndroid/ToolbarAndroid.android.js";

  function u(o, n) {
      var t = Object.keys(o);
      if (Object.getOwnPropertySymbols) {
          var c = Object.getOwnPropertySymbols(o);
          n && (c = c.filter(function(n) {
              return Object.getOwnPropertyDescriptor(o, n).enumerable
          })), t.push.apply(t, c)
      }
      return t
  }

  function v(o) {
      for (var t = 1; t < arguments.length; t++) {
          var c = null != arguments[t] ? arguments[t] : {};
          t % 2 ? u(c, !0).forEach(function(t) {
              n(o, t, c[t])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(o, Object.getOwnPropertyDescriptors(c)) : u(c).forEach(function(n) {
              Object.defineProperty(o, n, Object.getOwnPropertyDescriptor(c, n))
          })
      }
      return o
  }
  var w = r(d[8]),
      b = r(d[9]),
      y = r(d[10]),
      O = r(d[11]),
      A = (function(n) {
          function u() {
              var o, n;
              c(this, u);
              for (var t = arguments.length, s = new Array(t), f = 0; f < t; f++) s[f] = arguments[f];
              return (n = p(this, (o = l(u)).call.apply(o, [this].concat(s))))._onSelect = function(o) {
                  var t = o.nativeEvent.position; - 1 === t ? n.props.onIconClicked && n.props.onIconClicked() : n.props.onActionSelected && n.props.onActionSelected(t)
              }, n
          }
          return f(u, n), s(u, [{
              key: "render",
              value: function() {
                  var n = this.props,
                      c = (n.onIconClicked, n.onActionSelected, n.forwardedRef),
                      s = v({}, t(n, ["onIconClicked", "onActionSelected", "forwardedRef"]));
                  if (this.props.logo && (s.logo = O(this.props.logo)), this.props.navIcon && (s.navIcon = O(this.props.navIcon)), this.props.overflowIcon && (s.overflowIcon = O(this.props.overflowIcon)), this.props.actions) {
                      for (var p = [], l = 0; l < this.props.actions.length; l++) {
                          var f = {
                              icon: this.props.actions[l].icon,
                              show: this.props.actions[l].show
                          };
                          f.icon && (f.icon = O(f.icon)), f.show && (f.show = b.getViewManagerConfig('ToolbarAndroid').Constants.ShowAsAction[f.show]), p.push(v({}, this.props.actions[l], {}, f))
                      }
                      s.nativeActions = p
                  }
                  return w.createElement(y, o({
                      onSelect: this._onSelect
                  }, s, {
                      ref: c,
                      __source: {
                          fileName: h,
                          lineNumber: 228
                      }
                  }))
              }
          }]), u
      })(w.Component),
      j = w.forwardRef(function(n, t) {
          return w.createElement(A, o({}, n, {
              forwardedRef: t,
              __source: {
                  fileName: h,
                  lineNumber: 242
              }
          }))
      });
  m.exports = j
}
