// react-native/Libraries/Components/CheckBox/CheckBox.android.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      u = r(d[4]),
      c = r(d[5]),
      l = r(d[6]),
      f = r(d[7]),
      p = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/CheckBox/CheckBox.android.js";

  function h(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }

  function v(t) {
      for (var o = 1; o < arguments.length; o++) {
          var s = null != arguments[o] ? arguments[o] : {};
          o % 2 ? h(s, !0).forEach(function(o) {
              n(t, o, s[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(s)) : h(s).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(s, n))
          })
      }
      return t
  }
  var b = r(d[8]),
      y = r(d[9]),
      w = r(d[10]),
      O = r(d[11]),
      R = r(d[12]),
      _ = (function(n) {
          function h() {
              var t, n;
              s(this, h);
              for (var o = arguments.length, u = new Array(o), f = 0; f < o; f++) u[f] = arguments[f];
              return (n = c(this, (t = l(h)).call.apply(t, [this].concat(u))))._nativeRef = null, n._setNativeRef = R({
                  getForwardedRef: function() {
                      return n.props.forwardedRef
                  },
                  setLocalRef: function(t) {
                      n._nativeRef = t
                  }
              }), n._onChange = function(t) {
                  var o, s = null != (o = n.props.value) && o;
                  O(n._nativeRef).setNativeProps({
                      value: s
                  }), n.props.onChange && n.props.onChange(t), n.props.onValueChange && n.props.onValueChange(t.nativeEvent.value)
              }, n
          }
          return f(h, n), u(h, [{
              key: "render",
              value: function() {
                  var n, s, u = this.props,
                      c = (u.disabled, u.value, u.style),
                      l = (u.forwardedRef, v({}, o(u, ["disabled", "value", "style", "forwardedRef"]), {
                          onStartShouldSetResponder: function() {
                              return !0
                          },
                          onResponderTerminationRequest: function() {
                              return !1
                          },
                          enabled: !(null != (n = this.props.disabled) && n),
                          on: null != (s = this.props.value) && s,
                          style: [C.rctCheckBox, c]
                      }));
                  return b.createElement(w, t({}, l, {
                      ref: this._setNativeRef,
                      onChange: this._onChange,
                      __source: {
                          fileName: p,
                          lineNumber: 168
                      }
                  }))
              }
          }]), h
      })(b.Component),
      C = y.create({
          rctCheckBox: {
              height: 32,
              width: 32
          }
      }),
      j = b.forwardRef(function(n, o) {
          return b.createElement(_, t({}, n, {
              forwardedRef: o,
              __source: {
                  fileName: p,
                  lineNumber: 190
              }
          }))
      });
  m.exports = j
}
