// react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/ActivityIndicator/ActivityIndicator.js";

  function c(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }

  function l(t) {
      for (var o = 1; o < arguments.length; o++) {
          var s = null != arguments[o] ? arguments[o] : {};
          o % 2 ? c(s, !0).forEach(function(o) {
              n(t, o, s[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(s)) : c(s).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(s, n))
          })
      }
      return t
  }
  r(d[3]);
  var u = r(d[4]),
      y = r(d[5]),
      p = r(d[6]),
      f = (r(d[7]), r(d[8])),
      b = u.forwardRef(function(n, c) {
          var b, O = n.onLayout,
              v = n.style,
              w = o(n, ["onLayout", "style"]);
          switch (n.size) {
              case 'small':
                  b = h.sizeSmall;
                  break;
              case 'large':
                  b = h.sizeLarge;
                  break;
              default:
                  b = {
                      height: n.size,
                      width: n.size
                  }
          }
          var j = l({}, w, {
              ref: c,
              style: b,
              styleAttr: 'Normal',
              indeterminate: !0
          });
          return u.createElement(p, {
              onLayout: O,
              style: y.compose(h.container, v),
              __source: {
                  fileName: s,
                  lineNumber: 97
              }
          }, u.createElement(f, t({}, j, {
              __source: {
                  fileName: s,
                  lineNumber: 105
              }
          })))
      });
  b.displayName = 'ActivityIndicator', b.defaultProps = {
      animating: !0,
      color: null,
      hidesWhenStopped: !0,
      size: 'small'
  };
  var h = y.create({
      container: {
          alignItems: 'center',
          justifyContent: 'center'
      },
      sizeSmall: {
          width: 20,
          height: 20
      },
      sizeLarge: {
          width: 36,
          height: 36
      }
  });
  m.exports = b
}
