// react-native-elements/src/header/Header.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Header = void 0;
  var o = t(r(d[2])),
      l = t(r(d[3])),
      c = t(r(d[4])),
      u = t(r(d[5])),
      f = t(r(d[6])),
      p = t(r(d[7])),
      s = t(r(d[8])),
      y = t(r(d[9])),
      h = n(r(d[10])),
      b = t(r(d[11])),
      S = r(d[12]),
      C = r(d[13]),
      P = r(d[14]),
      O = t(r(d[15])),
      w = t(r(d[16])),
      v = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/header/Header.js";

  function T(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function k(t) {
      for (var n = 1; n < arguments.length; n++) {
          var o = null != arguments[n] ? arguments[n] : {};
          n % 2 ? T(o, !0).forEach(function(n) {
              (0, y.default)(t, n, o[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : T(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var _ = {
          left: 'flex-start',
          right: 'flex-end',
          center: 'center'
      },
      j = function(t) {
          var n = t.style,
              o = t.placement,
              l = t.children;
          return h.default.createElement(S.View, {
              style: S.StyleSheet.flatten([{
                  alignItems: _[o]
              }, n]),
              __source: {
                  fileName: v,
                  lineNumber: 25
              }
          }, null == l || !1 === l ? null : l.text ? (0, P.renderNode)(O.default, l.text, k({
              numberOfLines: 1
          }, l)) : l.icon ? (0, P.renderNode)(w.default, k({}, l, {
              name: l.icon,
              containerStyle: S.StyleSheet.flatten([{
                  alignItems: _[o]
              }, l.containerStyle])
          })) : (0, P.renderNode)(O.default, l))
      };
  j.propTypes = {
      placement: b.default.oneOf(['left', 'center', 'right']),
      style: C.ViewPropTypes.style,
      children: b.default.oneOfType([P.nodeType, b.default.node])
  };
  var N = (function(t) {
      function n() {
          return (0, c.default)(this, n), (0, f.default)(this, (0, p.default)(n).apply(this, arguments))
      }
      return (0, s.default)(n, t), (0, u.default)(n, [{
          key: "componentDidMount",
          value: function() {
              var t = this.props,
                  n = t.linearGradientProps,
                  o = t.ViewComponent;
              !n || g.Expo || o || console.error("You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}")
          }
      }, {
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.statusBarProps,
                  c = t.leftComponent,
                  u = t.centerComponent,
                  f = t.rightComponent,
                  p = t.leftContainerStyle,
                  s = t.centerContainerStyle,
                  y = t.rightContainerStyle,
                  b = t.backgroundColor,
                  C = t.backgroundImage,
                  P = t.backgroundImageStyle,
                  O = t.containerStyle,
                  w = t.placement,
                  T = t.barStyle,
                  k = t.children,
                  _ = t.linearGradientProps,
                  N = t.ViewComponent,
                  I = void 0 === N ? _ && g.Expo ? g.Expo.LinearGradient : S.ImageBackground : N,
                  V = t.theme,
                  x = (0, l.default)(t, ["statusBarProps", "leftComponent", "centerComponent", "rightComponent", "leftContainerStyle", "centerContainerStyle", "rightContainerStyle", "backgroundColor", "backgroundImage", "backgroundImageStyle", "containerStyle", "placement", "barStyle", "children", "linearGradientProps", "ViewComponent", "theme"]);
              return h.default.createElement(I, (0, o.default)({
                  testID: "headerContainer"
              }, x, {
                  style: S.StyleSheet.flatten([E.container(V), b && {
                      backgroundColor: b
                  }, O]),
                  source: C,
                  imageStyle: P
              }, _, {
                  __source: {
                      fileName: v,
                      lineNumber: 86
                  }
              }), h.default.createElement(S.StatusBar, (0, o.default)({
                  barStyle: T
              }, n, {
                  __source: {
                      fileName: v,
                      lineNumber: 98
                  }
              })), h.default.createElement(j, {
                  style: S.StyleSheet.flatten(['center' === w && E.rightLeftContainer, p]),
                  placement: "left",
                  __source: {
                      fileName: v,
                      lineNumber: 99
                  }
              }, h.default.isValidElement(k) && k || k[0] || c), h.default.createElement(j, {
                  style: S.StyleSheet.flatten([E.centerContainer, 'center' !== w && {
                      paddingHorizontal: S.Platform.select({
                          android: 16,
                          default: 15
                      })
                  }, s]),
                  placement: w,
                  __source: {
                      fileName: v,
                      lineNumber: 111
                  }
              }, k[1] || u), h.default.createElement(j, {
                  style: S.StyleSheet.flatten(['center' === w && E.rightLeftContainer, y]),
                  placement: "right",
                  __source: {
                      fileName: v,
                      lineNumber: 127
                  }
              }, k[2] || f))
          }
      }]), n
  })(h.Component);
  e.Header = N, N.propTypes = {
      placement: b.default.oneOf(['left', 'center', 'right']),
      leftComponent: P.nodeType,
      centerComponent: P.nodeType,
      rightComponent: P.nodeType,
      leftContainerStyle: C.ViewPropTypes.style,
      centerContainerStyle: C.ViewPropTypes.style,
      rightContainerStyle: C.ViewPropTypes.style,
      backgroundColor: b.default.string,
      backgroundImage: S.Image.propTypes.source,
      backgroundImageStyle: S.Image.propTypes.style,
      containerStyle: C.ViewPropTypes.style,
      statusBarProps: b.default.object,
      barStyle: b.default.oneOf(['default', 'light-content', 'dark-content']),
      children: b.default.oneOfType([b.default.arrayOf(b.default.node), b.default.node]),
      theme: b.default.object,
      linearGradientProps: b.default.object,
      ViewComponent: b.default.oneOfType([b.default.func, b.default.object])
  }, N.defaultProps = {
      placement: 'center',
      children: []
  };
  var E = {
          container: function(t) {
              return {
                  borderBottomColor: '#f2f2f2',
                  borderBottomWidth: S.StyleSheet.hairlineWidth,
                  paddingHorizontal: 10,
                  backgroundColor: t.colors.primary,
                  paddingTop: (0, C.getStatusBarHeight)(),
                  flexDirection: 'row',
                  justifyContent: 'space-between',
                  alignItems: 'center',
                  height: S.Platform.select({
                      android: 56,
                      default: 44
                  }) + (0, C.getStatusBarHeight)()
              }
          },
          centerContainer: {
              flex: 3
          },
          rightLeftContainer: {
              flex: 1
          }
      },
      I = (0, C.withTheme)(N, 'Header');
  e.default = I
}
