// react-native-gesture-handler/touchables/TouchableOpacity.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      c = t(r(d[5])),
      p = t(r(d[6])),
      s = t(r(d[7])),
      f = t(r(d[8])),
      y = t(r(d[9])),
      O = r(d[10]),
      h = n(r(d[11])),
      v = n(r(d[12])),
      b = t(r(d[13])),
      E = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-gesture-handler/touchables/TouchableOpacity.js";

  function T(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function S(t) {
      for (var n = 1; n < arguments.length; n++) {
          var l = null != arguments[n] ? arguments[n] : {};
          n % 2 ? T(l, !0).forEach(function(n) {
              (0, o.default)(t, n, l[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : T(l).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(l, n))
          })
      }
      return t
  }
  var D = (function(t) {
      function n() {
          var t, o;
          (0, c.default)(this, n);
          for (var l = arguments.length, u = new Array(l), p = 0; p < l; p++) u[p] = arguments[p];
          return (o = (0, s.default)(this, (t = (0, f.default)(n)).call.apply(t, [this].concat(u)))).getChildStyleOpacityWithDefault = function() {
              var t = O.StyleSheet.flatten(o.props.style) || {};
              return null == t.opacity ? 1 : t.opacity
          }, o.opacity = new O.Animated.Value(o.getChildStyleOpacityWithDefault()), o.setOpacityTo = function(t, n) {
              O.Animated.timing(o.opacity, {
                  toValue: t,
                  duration: n,
                  easing: O.Easing.inOut(O.Easing.quad),
                  useNativeDriver: !0
              }).start()
          }, o.onStateChange = function(t, n) {
              n === h.TOUCHABLE_STATE.BEGAN ? o.setOpacityTo(o.props.activeOpacity, 0) : n !== h.TOUCHABLE_STATE.UNDETERMINED && n !== h.TOUCHABLE_STATE.MOVED_OUTSIDE || o.setOpacityTo(o.getChildStyleOpacityWithDefault(), 150)
          }, o
      }
      return (0, y.default)(n, t), (0, p.default)(n, [{
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.style,
                  o = void 0 === n ? {} : n,
                  c = (0, u.default)(t, ["style"]);
              return v.default.createElement(h.default, (0, l.default)({}, c, {
                  style: [o, {
                      opacity: this.opacity
                  }],
                  onStateChange: this.onStateChange,
                  __source: {
                      fileName: E,
                      lineNumber: 52
                  }
              }), this.props.children ? this.props.children : v.default.createElement(O.View, {
                  __source: {
                      fileName: E,
                      lineNumber: 61
                  }
              }))
          }
      }]), n
  })(v.Component);
  e.default = D, D.defaultProps = S({}, h.default.defaultProps, {
      activeOpacity: .2
  }), D.propTypes = S({}, h.default.publicPropTypes, {
      style: b.default.any,
      activeOpacity: b.default.number
  })
}
