// react-native-gesture-handler/touchables/TouchableNativeFeedback.android.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      s = t(r(d[5])),
      c = t(r(d[6])),
      p = t(r(d[7])),
      f = t(r(d[8])),
      b = t(r(d[9])),
      y = t(r(d[10])),
      v = o(r(d[11])),
      P = t(r(d[12]));

  function O(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }

  function h(t) {
      for (var o = 1; o < arguments.length; o++) {
          var l = null != arguments[o] ? arguments[o] : {};
          o % 2 ? O(l, !0).forEach(function(o) {
              (0, n.default)(t, o, l[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : O(l).forEach(function(o) {
              Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(l, o))
          })
      }
      return t
  }
  var j = (function(t) {
      function o() {
          var t, n;
          (0, s.default)(this, o);
          for (var l = arguments.length, u = new Array(l), c = 0; c < l; c++) u[c] = arguments[c];
          return (n = (0, p.default)(this, (t = (0, f.default)(o)).call.apply(t, [this].concat(u)))).getExtraButtonProps = function() {
              var t = {},
                  o = n.props.background;
              return o && ('Ripple' === o.type ? (t.borderless = o.borderless, t.rippleColor = o.color) : 'SelectableBackgroundBorderless' === o.type && (t.borderless = !0)), t.foreground = n.props.useForeground, t
          }, n
      }
      return (0, b.default)(o, t), (0, c.default)(o, [{
          key: "render",
          value: function() {
              var t = this.props,
                  o = t.style,
                  n = void 0 === o ? {} : o,
                  s = (0, u.default)(t, ["style"]);
              return v.default.createElement(y.default, (0, l.default)({}, s, {
                  style: n,
                  extraButtonProps: this.getExtraButtonProps(),
                  __source: {
                      fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-gesture-handler/touchables/TouchableNativeFeedback.android.js",
                      lineNumber: 58
                  }
              }))
          }
      }]), o
  })(v.Component);
  e.default = j, j.SelectableBackground = function() {
      return {
          type: 'SelectableBackground'
      }
  }, j.SelectableBackgroundBorderless = function() {
      return {
          type: 'SelectableBackgroundBorderless'
      }
  }, j.Ripple = function(t, o) {
      return {
          type: 'Ripple',
          color: t,
          borderless: o
      }
  }, j.canUseNativeForeground = function() {
      return Platform.Version >= 23
  }, j.defaultProps = h({}, y.default.defaultProps, {
      useForeground: !0,
      extraButtonProps: {
          rippleColor: null
      }
  }), j.propTypes = h({}, y.default.publicPropTypes, {
      useForeground: P.default.bool,
      background: P.default.object,
      style: P.default.any
  })
}
