// react-native-elements/src/tile/FeaturedTile.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.FeaturedTile = void 0;
  var n = t(r(d[1])),
      l = t(r(d[2])),
      o = t(r(d[3])),
      c = t(r(d[4])),
      u = t(r(d[5])),
      f = r(d[6]),
      s = r(d[7]),
      y = r(d[8]),
      p = t(r(d[9])),
      S = t(r(d[10])),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/tile/FeaturedTile.js";

  function b(t, n) {
      var l = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var o = Object.getOwnPropertySymbols(t);
          n && (o = o.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), l.push.apply(l, o)
      }
      return l
  }

  function C(t) {
      for (var n = 1; n < arguments.length; n++) {
          var l = null != arguments[n] ? arguments[n] : {};
          n % 2 ? b(l, !0).forEach(function(n) {
              (0, o.default)(t, n, l[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : b(l).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(l, n))
          })
      }
      return t
  }
  var w = function(t) {
      var o, u, s, b = t.title,
          w = t.icon,
          v = t.caption,
          T = t.imageSrc,
          P = t.containerStyle,
          O = t.imageContainerStyle,
          j = t.overlayContainerStyle,
          _ = t.iconContainerStyle,
          N = t.titleStyle,
          I = t.captionStyle,
          E = t.ImageComponent,
          x = t.imageProps,
          D = (0, l.default)(t, ["title", "icon", "caption", "imageSrc", "containerStyle", "imageContainerStyle", "overlayContainerStyle", "iconContainerStyle", "titleStyle", "captionStyle", "ImageComponent", "imageProps"]),
          V = t.width,
          k = void 0 === V ? f.Dimensions.get('window').width : V,
          F = t.height,
          B = void 0 === F ? .8 * k : F,
          A = f.StyleSheet.create({
              container: {
                  width: k,
                  height: B
              },
              imageContainer: {
                  alignItems: 'center',
                  justifyContent: 'center',
                  width: k,
                  height: B
              },
              overlayContainer: {
                  flex: 1,
                  alignItems: 'center',
                  alignSelf: 'stretch',
                  justifyContent: 'center',
                  paddingLeft: 25,
                  paddingRight: 25,
                  paddingTop: 45,
                  paddingBottom: 40,
                  position: 'absolute',
                  top: 0,
                  left: 0,
                  right: 0,
                  bottom: 0
              },
              text: {
                  color: '#ffffff',
                  backgroundColor: 'rgba(0,0,0,0)',
                  marginBottom: 15,
                  textAlign: 'center'
              },
              iconContainer: {
                  justifyContent: 'center',
                  alignItems: 'center',
                  alignSelf: 'center'
              }
          });
      return c.default.createElement(f.TouchableOpacity, (0, n.default)({}, D, {
          style: f.StyleSheet.flatten([A.container, P && P]),
          __source: {
              fileName: h,
              lineNumber: 86
          }
      }), c.default.createElement(E, (0, n.default)({
          resizeMode: "cover"
      }, x, {
          source: T,
          style: f.StyleSheet.flatten([A.imageContainer, O && O]),
          __source: {
              fileName: h,
              lineNumber: 93
          }
      }), c.default.createElement(f.View, {
          style: f.StyleSheet.flatten([A.overlayContainer, j && j]),
          __source: {
              fileName: h,
              lineNumber: 102
          }
      }, c.default.createElement(f.View, {
          style: f.StyleSheet.flatten([A.iconContainer, _ && _]),
          __source: {
              fileName: h,
              lineNumber: 108
          }
      }, w && c.default.createElement(S.default, (0, n.default)({}, w, {
          __source: {
              fileName: h,
              lineNumber: 114
          }
      }))), c.default.createElement(p.default, {
          testID: "featuredTileTitle",
          h4: !0,
          style: f.StyleSheet.flatten([A.text, N && N]),
          __source: {
              fileName: h,
              lineNumber: 116
          }
      }, b), (o = v, u = {
          style: I
      }, s = A.text, (0, y.renderNode)(p.default, o, C({}, u, {
          style: f.StyleSheet.flatten([s, u && u.style])
      }))))))
  };
  e.FeaturedTile = w, w.propTypes = {
      title: u.default.string,
      icon: u.default.object,
      caption: u.default.node,
      imageSrc: f.Image.propTypes.source,
      onPress: u.default.func,
      containerStyle: s.ViewPropTypes.style,
      iconContainerStyle: s.ViewPropTypes.style,
      imageContainerStyle: s.ViewPropTypes.style,
      overlayContainerStyle: s.ViewPropTypes.style,
      titleStyle: f.Text.propTypes.style,
      captionStyle: f.Text.propTypes.style,
      width: u.default.number,
      height: u.default.number,
      ImageComponent: u.default.oneOfType([u.default.func, u.default.object]),
      imageProps: u.default.object
  }, w.defaultProps = {
      ImageComponent: s.BackgroundImage,
      imageProps: {}
  };
  var v = (0, s.withTheme)(w, 'FeaturedTile');
  e.default = v
}
