// react-native-elements/src/tile/Tile.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Tile = void 0;
  var n = t(r(d[1])),
      l = t(r(d[2])),
      o = t(r(d[3])),
      c = t(r(d[4])),
      u = t(r(d[5])),
      y = r(d[6]),
      s = r(d[7]),
      f = t(r(d[8])),
      p = t(r(d[9])),
      S = t(r(d[10])),
      b = t(r(d[11])),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/tile/Tile.js";

  function C(t, n) {
      var l = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var o = Object.getOwnPropertySymbols(t);
          n && (o = o.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), l.push.apply(l, o)
      }
      return l
  }

  function O(t) {
      for (var l = 1; l < arguments.length; l++) {
          var o = null != arguments[l] ? arguments[l] : {};
          l % 2 ? C(o, !0).forEach(function(l) {
              (0, n.default)(t, l, o[l])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : C(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var P = function(t) {
      var n = t.featured,
          u = t.imageSrc,
          s = t.icon,
          f = t.title,
          C = t.children,
          P = t.caption,
          w = t.titleStyle,
          T = t.onPress,
          _ = t.activeOpacity,
          j = t.overlayContainerStyle,
          N = t.captionStyle,
          E = t.iconContainerStyle,
          I = t.imageContainerStyle,
          D = t.containerStyle,
          V = t.contentContainerStyle,
          x = t.titleNumberOfLines,
          L = t.ImageComponent,
          k = t.imageProps,
          B = (0, o.default)(t, ["featured", "imageSrc", "icon", "title", "children", "caption", "titleStyle", "onPress", "activeOpacity", "overlayContainerStyle", "captionStyle", "iconContainerStyle", "imageContainerStyle", "containerStyle", "contentContainerStyle", "titleNumberOfLines", "ImageComponent", "imageProps"]),
          M = t.width,
          z = t.height,
          A = void 0 === z ? .8 * M : z;
      if (n) {
          var F = {
              title: f,
              icon: s,
              caption: P,
              imageSrc: u,
              onPress: T,
              activeOpacity: _,
              containerStyle: D,
              imageContainerStyle: I,
              overlayContainerStyle: j,
              titleStyle: w,
              captionStyle: N,
              width: M,
              height: A,
              imageProps: k,
              ImageComponent: L
          };
          return c.default.createElement(b.default, (0, l.default)({}, F, {
              __source: {
                  fileName: h,
                  lineNumber: 61
              }
          }))
      }
      return c.default.createElement(y.TouchableOpacity, (0, l.default)({}, B, {
          onPress: T,
          activeOpacity: _,
          style: y.StyleSheet.flatten([{
              width: M,
              height: A
          }, D && D]),
          __source: {
              fileName: h,
              lineNumber: 65
          }
      }), c.default.createElement(L, (0, l.default)({
          resizeMode: "cover",
          source: u,
          containerStyle: y.StyleSheet.flatten([v.imageContainer, I && I]),
          style: O({}, y.StyleSheet.absoluteFillObject, {
              alignItems: 'center',
              justifyContent: 'center'
          })
      }, k, {
          __source: {
              fileName: h,
              lineNumber: 77
          }
      }), c.default.createElement(y.View, {
          style: y.StyleSheet.flatten([v.iconContainer, E && E]),
          __source: {
              fileName: h,
              lineNumber: 91
          }
      }, s && c.default.createElement(S.default, (0, l.default)({}, s, {
          __source: {
              fileName: h,
              lineNumber: 97
          }
      })))), c.default.createElement(y.View, {
          style: y.StyleSheet.flatten([v.contentContainer, V && V]),
          __source: {
              fileName: h,
              lineNumber: 101
          }
      }, c.default.createElement(p.default, {
          testID: "tileTitle",
          h4: !0,
          style: y.StyleSheet.flatten([v.text, w && w]),
          numberOfLines: x,
          __source: {
              fileName: h,
              lineNumber: 107
          }
      }, f), C))
  };
  e.Tile = P, P.propTypes = {
      title: u.default.string,
      icon: u.default.object,
      caption: u.default.node,
      imageSrc: y.Image.propTypes.source,
      onPress: u.default.func,
      activeOpacity: u.default.number,
      containerStyle: s.ViewPropTypes.style,
      imageContainerStyle: s.ViewPropTypes.style,
      iconContainerStyle: s.ViewPropTypes.style,
      overlayContainerStyle: s.ViewPropTypes.style,
      titleStyle: s.TextPropTypes.style,
      captionStyle: s.TextPropTypes.style,
      width: u.default.number,
      height: u.default.number,
      featured: u.default.bool,
      children: u.default.node,
      contentContainerStyle: s.ViewPropTypes.style,
      titleNumberOfLines: u.default.number,
      imageProps: u.default.object,
      ImageComponent: u.default.oneOfType([u.default.func, u.default.object])
  }, P.defaultProps = {
      width: y.Dimensions.get('window').width,
      ImageComponent: f.default,
      imageProps: {}
  };
  var v = y.StyleSheet.create({
          imageContainer: {
              flex: 2
          },
          text: {
              backgroundColor: 'rgba(0,0,0,0)',
              marginBottom: 5
          },
          contentContainer: {
              paddingTop: 15,
              paddingBottom: 5,
              paddingLeft: 15,
              paddingRight: 15
          },
          iconContainer: {
              justifyContent: 'center',
              alignItems: 'center',
              alignSelf: 'center'
          }
      }),
      w = (0, s.withTheme)(P, 'Tile');
  e.default = w
}
