// react-native/Libraries/Components/DatePicker/DatePickerIOS.android.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      n = r(d[1]),
      o = r(d[2]),
      s = r(d[3]),
      c = r(d[4]),
      u = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/DatePicker/DatePickerIOS.android.js",
      l = r(d[5]),
      p = r(d[6]),
      f = r(d[7]),
      k = r(d[8]),
      h = (function(p) {
          function h() {
              return t(this, h), o(this, s(h).apply(this, arguments))
          }
          return c(h, p), n(h, [{
              key: "render",
              value: function() {
                  return l.createElement(k, {
                      style: [y.dummyDatePickerIOS, this.props.style],
                      __source: {
                          fileName: u,
                          lineNumber: 20
                      }
                  }, l.createElement(f, {
                      style: y.datePickerText,
                      __source: {
                          fileName: u,
                          lineNumber: 21
                      }
                  }, "DatePickerIOS is not supported on this platform!"))
              }
          }]), h
      })(l.Component),
      y = p.create({
          dummyDatePickerIOS: {
              height: 100,
              width: 300,
              backgroundColor: '#ffbcbc',
              borderWidth: 1,
              borderColor: 'red',
              alignItems: 'center',
              justifyContent: 'center',
              margin: 10
          },
          datePickerText: {
              color: '#333333',
              margin: 20
          }
      });
  m.exports = h
}
