// react-native-elements/src/buttons/Button.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Button = void 0;
  var n = t(r(d[2])),
      l = t(r(d[3])),
      s = t(r(d[4])),
      u = t(r(d[5])),
      c = t(r(d[6])),
      f = t(r(d[7])),
      p = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      h = o(r(d[11])),
      S = r(d[12]),
      P = r(d[13]),
      w = r(d[14]),
      T = t(r(d[15])),
      v = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/buttons/Button.js";

  function C(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }

  function O(t) {
      for (var o = 1; o < arguments.length; o++) {
          var n = null != arguments[o] ? arguments[o] : {};
          o % 2 ? C(n, !0).forEach(function(o) {
              (0, l.default)(t, o, n[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : C(n).forEach(function(o) {
              Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
          })
      }
      return t
  }
  var j = function(t, o) {
          return {
              color: 'solid' === t ? 'white' : o.colors.primary,
              size: 'small'
          }
      },
      k = (function(t) {
          function o() {
              return (0, u.default)(this, o), (0, f.default)(this, (0, p.default)(o).apply(this, arguments))
          }
          return (0, y.default)(o, t), (0, c.default)(o, [{
              key: "componentDidMount",
              value: function() {
                  var t = this.props,
                      o = t.linearGradientProps,
                      n = t.ViewComponent;
                  !o || g.Expo || n || console.error("You need to pass a ViewComponent to use linearGradientProps !\nExample: ViewComponent={require('react-native-linear-gradient')}")
              }
          }, {
              key: "render",
              value: function() {
                  var t = this.props,
                      o = t.TouchableComponent,
                      l = t.containerStyle,
                      u = t.onPress,
                      c = t.buttonStyle,
                      f = t.type,
                      p = t.loading,
                      y = t.loadingStyle,
                      b = t.loadingProps,
                      P = t.title,
                      C = t.titleProps,
                      k = t.titleStyle,
                      V = t.icon,
                      _ = t.iconContainerStyle,
                      R = t.iconRight,
                      E = t.disabled,
                      x = t.disabledStyle,
                      D = t.disabledTitleStyle,
                      z = t.raised,
                      B = t.linearGradientProps,
                      G = t.ViewComponent,
                      F = void 0 === G ? !E && B && g.Expo ? g.Expo.LinearGradient : S.View : G,
                      A = t.theme,
                      I = (0, s.default)(t, ["TouchableComponent", "containerStyle", "onPress", "buttonStyle", "type", "loading", "loadingStyle", "loadingProps", "title", "titleProps", "titleStyle", "icon", "iconContainerStyle", "iconRight", "disabled", "disabledStyle", "disabledTitleStyle", "raised", "linearGradientProps", "ViewComponent", "theme"]);
                  'android' === S.Platform.OS && c.borderRadius && !I.background && (S.Platform.Version >= 21 ? I.background = S.TouchableNativeFeedback.Ripple(void 0, !1) : I.background = S.TouchableNativeFeedback.SelectableBackground());
                  var M = O({}, j(f, A), {}, b);
                  return h.default.createElement(S.View, {
                      style: S.StyleSheet.flatten([N.container, {
                          borderRadius: c.borderRadius || N.container.borderRadius
                      }, l, z && !E && N.raised(f)]),
                      __source: {
                          fileName: v,
                          lineNumber: 80
                      }
                  }, h.default.createElement(o, (0, n.default)({
                      onPress: u,
                      activeOpacity: .3,
                      disabled: E
                  }, I, {
                      __source: {
                          fileName: v,
                          lineNumber: 91
                      }
                  }), h.default.createElement(F, (0, n.default)({}, B, {
                      style: S.StyleSheet.flatten([N.button(f, A), c, E && N.disabled(f, A), E && x]),
                      __source: {
                          fileName: v,
                          lineNumber: 97
                      }
                  }), p && h.default.createElement(S.ActivityIndicator, (0, n.default)({
                      style: S.StyleSheet.flatten([N.loading, y]),
                      color: M.color,
                      size: M.size
                  }, M, {
                      __source: {
                          fileName: v,
                          lineNumber: 107
                      }
                  })), !p && V && !R && (0, w.renderNode)(T.default, V, {
                      containerStyle: S.StyleSheet.flatten([N.iconContainer, _])
                  }), !p && !!P && h.default.createElement(S.Text, (0, n.default)({
                      style: S.StyleSheet.flatten([N.title(f, A), k, E && N.disabledTitle(A), E && D])
                  }, C, {
                      __source: {
                          fileName: v,
                          lineNumber: 126
                      }
                  }), P), !p && V && R && (0, w.renderNode)(T.default, V, {
                      containerStyle: S.StyleSheet.flatten([N.iconContainer, _])
                  }))))
              }
          }]), o
      })(h.Component);
  e.Button = k, k.propTypes = {
      title: b.default.string,
      titleStyle: S.Text.propTypes.style,
      titleProps: b.default.object,
      buttonStyle: P.ViewPropTypes.style,
      type: b.default.oneOf(['solid', 'clear', 'outline']),
      loading: b.default.bool,
      loadingStyle: P.ViewPropTypes.style,
      loadingProps: b.default.object,
      onPress: b.default.func,
      containerStyle: P.ViewPropTypes.style,
      icon: w.nodeType,
      iconContainerStyle: P.ViewPropTypes.style,
      iconRight: b.default.bool,
      linearGradientProps: b.default.object,
      TouchableComponent: b.default.oneOfType([b.default.func, b.default.object]),
      ViewComponent: b.default.oneOfType([b.default.func, b.default.object]),
      disabled: b.default.bool,
      disabledStyle: P.ViewPropTypes.style,
      disabledTitleStyle: S.Text.propTypes.style,
      raised: b.default.bool,
      theme: b.default.object
  }, k.defaultProps = {
      title: '',
      iconRight: !1,
      TouchableComponent: S.Platform.select({
          android: S.TouchableNativeFeedback,
          default: S.TouchableOpacity
      }),
      onPress: function() {
          return console.log('Please attach a method to this component')
      },
      type: 'solid',
      buttonStyle: {
          borderRadius: 3
      },
      disabled: !1,
      raised: !1,
      loading: !1
  };
  var N = {
          button: function(t, o) {
              return {
                  flexDirection: 'row',
                  justifyContent: 'center',
                  alignItems: 'center',
                  borderRadius: 3,
                  backgroundColor: 'solid' === t ? o.colors.primary : 'transparent',
                  padding: 8,
                  borderWidth: 'outline' === t ? S.StyleSheet.hairlineWidth : 0,
                  borderColor: o.colors.primary
              }
          },
          container: {
              borderRadius: 3
          },
          disabled: function(t, o) {
              return O({}, (0, w.conditionalStyle)('solid' === t, {
                  backgroundColor: o.colors.disabled
              }), {}, (0, w.conditionalStyle)('outline' === t, {
                  borderColor: (0, w.color)(o.colors.disabled).darken(.3)
              }))
          },
          disabledTitle: function(t) {
              return {
                  color: (0, w.color)(t.colors.disabled).darken(.3)
              }
          },
          title: function(t, o) {
              return O({
                  color: 'solid' === t ? 'white' : o.colors.primary,
                  fontSize: 16,
                  textAlign: 'center',
                  paddingTop: 2,
                  paddingBottom: 1
              }, S.Platform.select({
                  android: {
                      fontFamily: 'sans-serif-medium'
                  },
                  default: {
                      fontSize: 18
                  }
              }))
          },
          iconContainer: {
              marginHorizontal: 5
          },
          raised: function(t) {
              return 'clear' !== t && O({
                  backgroundColor: '#fff'
              }, S.Platform.select({
                  android: {
                      elevation: 4
                  },
                  default: {
                      shadowColor: 'rgba(0,0,0, .4)',
                      shadowOffset: {
                          height: 1,
                          width: 1
                      },
                      shadowOpacity: 1,
                      shadowRadius: 1
                  }
              }))
          },
          loading: {
              marginVertical: 2
          }
      },
      V = (0, P.withTheme)(k, 'Button');
  e.default = V
}
