// react-native-onboarding-swiper/src/Dot.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[1])),
      n = t(r(d[2])),
      c = r(d[3]),
      s = t(r(d[4]));

  function u(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var c = Object.getOwnPropertySymbols(t);
          o && (c = c.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, c)
      }
      return n
  }

  function l(t) {
      for (var n = 1; n < arguments.length; n++) {
          var c = null != arguments[n] ? arguments[n] : {};
          n % 2 ? u(c, !0).forEach(function(n) {
              (0, o.default)(t, n, c[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(c)) : u(c).forEach(function(o) {
              Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(c, o))
          })
      }
      return t
  }
  var f = function(t) {
      var o, s = t.isLight,
          u = t.selected;
      return o = s ? u ? 'rgba(0, 0, 0, 0.8)' : 'rgba(0, 0, 0, 0.3)' : u ? '#fff' : 'rgba(255, 255, 255, 0.5)', n.default.createElement(c.View, {
          style: l({}, b.dot, {
              backgroundColor: o
          }),
          __source: {
              fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-onboarding-swiper/src/Dot.js",
              lineNumber: 13
          }
      })
  };
  f.propTypes = {
      isLight: s.default.bool.isRequired,
      selected: s.default.bool.isRequired
  };
  var b = {
          dot: {
              width: 6,
              height: 6,
              borderRadius: 3,
              marginHorizontal: 3
          }
      },
      p = f;
  e.default = p
}
