// react-native-onboarding-swiper/src/Dots.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[1])),
      u = t(r(d[2])),
      l = r(d[3]),
      o = t(r(d[4])),
      s = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-onboarding-swiper/src/Dots.js",
      f = function(t) {
          var o = t.isLight,
              f = t.numPages,
              _ = t.currentPage,
              p = t.Dot;
          return u.default.createElement(l.View, {
              style: c.container,
              __source: {
                  fileName: s,
                  lineNumber: 6
              }
          }, (0, n.default)(Array(f)).map(function(t, n) {
              return u.default.createElement(p, {
                  key: n,
                  selected: n === _,
                  isLight: o,
                  __source: {
                      fileName: s,
                      lineNumber: 8
                  }
              })
          }))
      };
  f.propTypes = {
      isLight: o.default.bool.isRequired,
      numPages: o.default.number.isRequired,
      currentPage: o.default.number.isRequired,
      Dot: o.default.oneOfType([o.default.element, o.default.func]).isRequired
  };
  var c = {
          container: {
              flex: 0,
              flexDirection: 'row',
              alignItems: 'center'
          }
      },
      _ = f;
  e.default = _
}
