// react-native-onboarding-swiper/src/Pagination.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[1])),
      o = r(d[2]),
      l = t(r(d[3])),
      u = t(r(d[4])),
      f = t(r(d[5])),
      s = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-onboarding-swiper/src/Pagination.js";

  function c(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function p(t) {
      for (var o = 1; o < arguments.length; o++) {
          var l = null != arguments[o] ? arguments[o] : {};
          o % 2 ? c(l, !0).forEach(function(o) {
              (0, n.default)(t, o, l[o])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : c(l).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(l, n))
          })
      }
      return t
  }
  var b = function(t) {
      var n = t.numPages,
          l = t.currentPage,
          c = t.isLight,
          b = t.bottomBarHeight,
          h = t.bottomBarColor,
          w = t.controlStatusBar,
          O = t.showSkip,
          S = t.showNext,
          N = t.showDone,
          P = t.onNext,
          _ = t.onSkip,
          R = t.onDone,
          k = t.skipLabel,
          q = t.nextLabel,
          x = t.allowFontScaling,
          B = t.SkipButtonComponent,
          D = t.NextButtonComponent,
          L = t.DoneButtonComponent,
          j = t.DotComponent,
          v = l + 1 === n,
          C = O && !v && u.default.createElement(B, {
              isLight: c,
              skipLabel: k,
              allowFontScaling: x,
              onPress: function() {
                  'function' == typeof _ && (w && o.StatusBar.setBarStyle('default', !0), _())
              },
              __source: {
                  fileName: s,
                  lineNumber: 32
              }
          }),
          E = S && !v && u.default.createElement(D, {
              nextLabel: q,
              allowFontScaling: x,
              isLight: c,
              onPress: P,
              __source: {
                  fileName: s,
                  lineNumber: 49
              }
          }),
          T = N && v && u.default.createElement(L, {
              isLight: c,
              allowFontScaling: x,
              onPress: function() {
                  'function' == typeof R && (w && o.StatusBar.setBarStyle('default', !0), R())
              },
              __source: {
                  fileName: s,
                  lineNumber: 59
              }
          });
      return u.default.createElement(o.View, {
          style: p({
              height: b,
              backgroundColor: h
          }, y.container),
          __source: {
              fileName: s,
              lineNumber: 74
          }
      }, u.default.createElement(o.View, {
          style: y.buttonLeft,
          __source: {
              fileName: s,
              lineNumber: 81
          }
      }, C), u.default.createElement(f.default, {
          isLight: c,
          numPages: n,
          currentPage: l,
          Dot: j,
          style: y.dots,
          __source: {
              fileName: s,
              lineNumber: 82
          }
      }), u.default.createElement(o.View, {
          style: y.buttonRight,
          __source: {
              fileName: s,
              lineNumber: 89
          }
      }, E, T))
  };
  b.propTypes = {
      numPages: l.default.number.isRequired,
      currentPage: l.default.number.isRequired,
      isLight: l.default.bool.isRequired,
      bottomBarHeight: l.default.number.isRequired,
      bottomBarColor: l.default.string.isRequired,
      showNext: l.default.bool.isRequired,
      showSkip: l.default.bool.isRequired,
      showDone: l.default.bool.isRequired,
      onNext: l.default.func.isRequired,
      onSkip: l.default.func,
      onDone: l.default.func,
      allowFontScaling: l.default.bool,
      skipLabel: l.default.oneOfType([l.default.element, l.default.string]).isRequired,
      nextLabel: l.default.oneOfType([l.default.element, l.default.string]).isRequired,
      SkipButtonComponent: l.default.oneOfType([l.default.element, l.default.func]).isRequired,
      DoneButtonComponent: l.default.oneOfType([l.default.element, l.default.func]).isRequired,
      NextButtonComponent: l.default.oneOfType([l.default.element, l.default.func]).isRequired,
      DotComponent: l.default.oneOfType([l.default.element, l.default.func]).isRequired
  };
  var y = {
          container: {
              paddingHorizontal: 0,
              flexDirection: 'row',
              justifyContent: 'space-between',
              alignItems: 'center'
          },
          buttonLeft: {
              width: 200,
              flexShrink: 1,
              alignItems: 'flex-start'
          },
          buttonRight: {
              width: 200,
              flexShrink: 1,
              alignItems: 'flex-end'
          },
          dots: {
              flexShrink: 0
          }
      },
      h = b;
  e.default = h
}
