// @react-navigation/native/src/Scrollables.js
module.exports = function(g, r, i, a, m, e, d) {
  var t, n = r(d[0]),
      o = n(r(d[1])),
      l = n(r(d[2])),
      f = n(r(d[3])),
      u = r(d[4]),
      s = r(d[5]),
      c = n(r(d[6])),
      p = n(r(d[7])),
      b = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/@react-navigation/native/src/Scrollables.js";

  function O(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          n && (l = l.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, l)
      }
      return o
  }

  function h(t) {
      for (var n = 1; n < arguments.length; n++) {
          var o = null != arguments[n] ? arguments[n] : {};
          n % 2 ? O(o, !0).forEach(function(n) {
              (0, l.default)(t, n, o[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : O(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }

  function w(t) {
      var n = t.onRefresh;
      return n ? ((0, p.default)('boolean' == typeof t.refreshing, '`refreshing` prop must be set as a boolean in order to use `onRefresh`, but got `' + JSON.stringify(t.refreshing) + '`'), h({}, t, {
          refreshControl: null == t.refreshControl ? f.default.createElement(u.RefreshControl, {
              refreshing: t.refreshing,
              onRefresh: n,
              progressViewOffset: t.progressViewOffset,
              __source: {
                  fileName: b,
                  lineNumber: 36
              }
          }) : t.refreshControl
      })) : t
  }
  t = 'android' === u.Platform.OS ? (0, c.default)(u.ScrollView) : (0, c.default)(s.ScrollView);
  var y = f.default.forwardRef(function(n, l) {
          return f.default.createElement(u.FlatList, (0, o.default)({
              ref: l
          }, n, {
              renderScrollComponent: function(n) {
                  return f.default.createElement(t, (0, o.default)({}, w(n), {
                      __source: {
                          fileName: b,
                          lineNumber: 55
                      }
                  }))
              },
              __source: {
                  fileName: b,
                  lineNumber: 51
              }
          }))
      }),
      _ = f.default.forwardRef(function(n, l) {
          return f.default.createElement(u.SectionList, (0, o.default)({
              ref: l
          }, n, {
              renderScrollComponent: function(n) {
                  return f.default.createElement(t, (0, o.default)({}, w(n), {
                      __source: {
                          fileName: b,
                          lineNumber: 65
                      }
                  }))
              },
              __source: {
                  fileName: b,
                  lineNumber: 61
              }
          }))
      });
  m.exports = {
      ScrollView: t,
      FlatList: y,
      SectionList: _
  }
}
