// @react-navigation/native/src/Themed.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var l = t(r(d[2])),
      u = t(r(d[3])),
      o = t(r(d[4])),
      s = t(r(d[5])),
      f = t(r(d[6])),
      c = t(r(d[7])),
      p = t(r(d[8])),
      h = n(r(d[9])),
      y = r(d[10]),
      x = r(d[11]),
      T = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/@react-navigation/native/src/Themed.js",
      v = (function(t) {
          function n() {
              return (0, o.default)(this, n), (0, f.default)(this, (0, c.default)(n).apply(this, arguments))
          }
          return (0, p.default)(n, t), (0, s.default)(n, [{
              key: "render",
              value: function() {
                  return h.createElement(y.Text, (0, u.default)({}, this.props, {
                      style: [{
                          color: x.ThemeColors[this.context].label
                      }, this.props.style],
                      __source: {
                          fileName: T,
                          lineNumber: 10
                      }
                  }))
              }
          }]), n
      })(h.Component);
  v.contextType = x.ThemeContext;
  var b = (function(t) {
      function n() {
          return (0, o.default)(this, n), (0, f.default)(this, (0, c.default)(n).apply(this, arguments))
      }
      return (0, p.default)(n, t), (0, s.default)(n, [{
          key: "render",
          value: function() {
              return h.createElement(y.TextInput, (0, u.default)({}, this.props, {
                  placeholderTextColor: 'dark' === this.context ? '#ebebf54c' : '#3c3c434c',
                  style: [{
                      color: x.ThemeColors[this.context].label
                  }, this.props.style],
                  __source: {
                      fileName: T,
                      lineNumber: 23
                  }
              }))
          }
      }]), n
  })(h.Component);
  b.contextType = x.ThemeContext;
  var _ = (function(t) {
      function n() {
          return (0, o.default)(this, n), (0, f.default)(this, (0, c.default)(n).apply(this, arguments))
      }
      return (0, p.default)(n, t), (0, s.default)(n, [{
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.barStyle,
                  o = (0, l.default)(t, ["barStyle"]);
              return h.createElement(y.StatusBar, (0, u.default)({
                  barStyle: n || ('dark' === this.context ? 'light-content' : 'default')
              }, o, {
                  __source: {
                      fileName: T,
                      lineNumber: 41
                  }
              }))
          }
      }]), n
  })(h.Component);
  _.contextType = x.ThemeContext;
  var C = {
      Text: v,
      StatusBar: _,
      TextInput: b
  };
  e.default = C
}
