// react-native-maps/lib/components/MapCalloutSubview.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      u = t(r(d[3])),
      l = t(r(d[4])),
      p = t(r(d[5])),
      s = t(r(d[6])),
      c = t(r(d[7])),
      f = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      O = r(d[11]),
      v = o(r(d[12]));

  function P(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var u = Object.getOwnPropertySymbols(t);
          o && (u = u.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, u)
      }
      return n
  }
  var w = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? P(n, !0).forEach(function(o) {
                  (0, f.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : P(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, O.ViewPropTypes || O.View.propTypes, {
          onPress: y.default.func
      }),
      h = (function(t) {
          function o() {
              return (0, u.default)(this, o), (0, p.default)(this, (0, s.default)(o).apply(this, arguments))
          }
          return (0, c.default)(o, t), (0, l.default)(o, [{
              key: "render",
              value: function() {
                  var t = this.getAirComponent();
                  return b.default.createElement(t, (0, n.default)({}, this.props, {
                      style: [j.calloutSubview, this.props.style],
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapCalloutSubview.js",
                          lineNumber: 23
                      }
                  }))
              }
          }]), o
      })(b.default.Component);
  h.propTypes = w, h.defaultProps = {};
  var j = O.StyleSheet.create({
          calloutSubview: {}
      }),
      S = (0, v.default)(h, {
          componentType: 'CalloutSubview',
          providers: {
              google: {
                  ios: v.SUPPORTED,
                  android: v.NOT_SUPPORTED
              }
          }
      });
  e.default = S
}
