// react-native-maps/lib/components/MapCircle.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      u = t(r(d[3])),
      l = t(r(d[4])),
      s = t(r(d[5])),
      f = t(r(d[6])),
      p = t(r(d[7])),
      c = t(r(d[8])),
      b = t(r(d[9])),
      y = t(r(d[10])),
      O = r(d[11]),
      P = o(r(d[12]));

  function v(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var u = Object.getOwnPropertySymbols(t);
          o && (u = u.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, u)
      }
      return n
  }
  var h = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? v(n, !0).forEach(function(o) {
                  (0, c.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : v(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, O.ViewPropTypes || O.View.propTypes, {
          center: b.default.shape({
              latitude: b.default.number.isRequired,
              longitude: b.default.number.isRequired
          }).isRequired,
          radius: b.default.number.isRequired,
          onPress: b.default.func,
          strokeWidth: b.default.number,
          strokeColor: O.ColorPropType,
          fillColor: O.ColorPropType,
          zIndex: b.default.number,
          lineCap: b.default.oneOf(['butt', 'round', 'square']),
          lineJoin: b.default.oneOf(['miter', 'round', 'bevel']),
          miterLimit: b.default.number,
          lineDashPhase: b.default.number,
          lineDashPattern: b.default.arrayOf(b.default.number)
      }),
      j = (function(t) {
          function o() {
              return (0, u.default)(this, o), (0, s.default)(this, (0, f.default)(o).apply(this, arguments))
          }
          return (0, p.default)(o, t), (0, l.default)(o, [{
              key: "setNativeProps",
              value: function(t) {
                  this.circle.setNativeProps(t)
              }
          }, {
              key: "render",
              value: function() {
                  var t = this,
                      o = this.getAirComponent();
                  return y.default.createElement(o, (0, n.default)({}, this.props, {
                      ref: function(o) {
                          t.circle = o
                      },
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapCircle.js",
                          lineNumber: 129
                      }
                  }))
              }
          }]), o
      })(y.default.Component);
  j.propTypes = h, j.defaultProps = {
      strokeColor: '#000',
      strokeWidth: 1
  };
  var w = (0, P.default)(j, {
      componentType: 'Circle',
      providers: {
          google: {
              ios: P.SUPPORTED,
              android: P.USES_DEFAULT_IMPLEMENTATION
          }
      }
  });
  e.default = w
}
