// react-native-maps/lib/components/MapCallout.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      p = t(r(d[5])),
      s = t(r(d[6])),
      f = t(r(d[7])),
      c = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      O = r(d[11]),
      P = o(r(d[12]));

  function v(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }
  var h = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? v(n, !0).forEach(function(o) {
                  (0, c.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : v(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, O.ViewPropTypes || O.View.propTypes, {
          tooltip: y.default.bool,
          onPress: y.default.func,
          alphaHitTest: y.default.bool
      }),
      j = (function(t) {
          function o() {
              return (0, l.default)(this, o), (0, p.default)(this, (0, s.default)(o).apply(this, arguments))
          }
          return (0, f.default)(o, t), (0, u.default)(o, [{
              key: "render",
              value: function() {
                  var t = this.getAirComponent();
                  return b.default.createElement(t, (0, n.default)({}, this.props, {
                      style: [w.callout, this.props.style],
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapCallout.js",
                          lineNumber: 28
                      }
                  }))
              }
          }]), o
      })(b.default.Component);
  j.propTypes = h, j.defaultProps = {
      tooltip: !1,
      alphaHitTest: !1
  };
  var w = O.StyleSheet.create({
          callout: {
              position: 'absolute'
          }
      }),
      T = (0, P.default)(j, {
          componentType: 'Callout',
          providers: {
              google: {
                  ios: P.SUPPORTED,
                  android: P.USES_DEFAULT_IMPLEMENTATION
              }
          }
      });
  e.default = T
}
