// react-native-maps/lib/components/MapHeatmap.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      u = t(r(d[3])),
      s = t(r(d[4])),
      l = t(r(d[5])),
      p = t(r(d[6])),
      f = t(r(d[7])),
      c = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      O = r(d[11]),
      h = o(r(d[12]));

  function v(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var u = Object.getOwnPropertySymbols(t);
          o && (u = u.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, u)
      }
      return n
  }
  var P = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? v(n, !0).forEach(function(o) {
                  (0, c.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : v(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, O.ViewPropTypes || O.View.propTypes, {
          points: y.default.arrayOf(y.default.shape({
              latitude: y.default.number.isRequired,
              longitude: y.default.number.isRequired,
              weight: y.default.number
          })),
          radius: y.default.number,
          opacity: y.default.number,
          gradient: y.default.shape({
              colors: y.default.arrayOf(y.default.string),
              startPoints: y.default.arrayOf(y.default.number),
              colorMapSize: y.default.number
          })
      }),
      j = (function(t) {
          function o() {
              return (0, u.default)(this, o), (0, l.default)(this, (0, p.default)(o).apply(this, arguments))
          }
          return (0, f.default)(o, t), (0, s.default)(o, [{
              key: "setNativeProps",
              value: function(t) {
                  this.heatmap.setNativeProps(t)
              }
          }, {
              key: "render",
              value: function() {
                  var t, o = this,
                      u = this.getAirComponent();
                  return this.props.gradient && ((t = (0, n.default)({}, this.props.gradient)).colors = t.colors.map(function(t) {
                      return (0, O.processColor)(t)
                  })), b.default.createElement(u, (0, n.default)({}, this.props, {
                      gradient: t,
                      ref: function(t) {
                          o.heatmap = t
                      },
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapHeatmap.js",
                          lineNumber: 79
                      }
                  }))
              }
          }]), o
      })(b.default.Component);
  j.propTypes = P, j.defaultProps = {
      strokeColor: '#000',
      strokeWidth: 1
  };
  var w = (0, h.default)(j, {
      componentType: 'Heatmap',
      providers: {
          google: {
              ios: h.SUPPORTED,
              android: h.USES_DEFAULT_IMPLEMENTATION
          }
      }
  });
  e.default = w
}
