// react-native-maps/lib/components/MapOverlay.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      s = t(r(d[3])),
      p = t(r(d[4])),
      u = t(r(d[5])),
      l = t(r(d[6])),
      f = t(r(d[7])),
      c = t(r(d[8])),
      y = o(r(d[9])),
      v = t(r(d[10])),
      O = r(d[11]),
      b = o(r(d[12]));

  function h(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          o && (s = s.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, s)
      }
      return n
  }
  var P = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? h(n, !0).forEach(function(o) {
                  (0, c.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : h(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, O.View.propTypes, {
          image: v.default.any.isRequired,
          bounds: v.default.arrayOf(v.default.array.isRequired).isRequired,
          tappable: v.default.bool,
          onPress: v.default.func
      }),
      j = (function(t) {
          function o() {
              return (0, s.default)(this, o), (0, u.default)(this, (0, l.default)(o).apply(this, arguments))
          }
          return (0, f.default)(o, t), (0, p.default)(o, [{
              key: "render",
              value: function() {
                  var t;
                  this.props.image && (t = 'function' == typeof this.props.image.startsWith && this.props.image.startsWith('http') ? this.props.image : (t = O.Image.resolveAssetSource(this.props.image) || {}).uri);
                  var o = this.getAirComponent();
                  return y.default.createElement(o, (0, n.default)({}, this.props, {
                      image: t,
                      style: [w.overlay, this.props.style],
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapOverlay.js",
                          lineNumber: 49
                      }
                  }))
              }
          }]), o
      })(y.Component);
  j.propTypes = P, j.viewConfig = {
      uiViewClassName: 'AIR<provider>MapOverlay',
      validAttributes: {
          image: !0
      }
  };
  var w = O.StyleSheet.create({
      overlay: {
          position: 'absolute',
          backgroundColor: 'transparent'
      }
  });
  j.Animated = O.Animated.createAnimatedComponent(j);
  var A = (0, b.default)(j, {
      componentType: 'Overlay',
      providers: {
          google: {
              ios: b.SUPPORTED,
              android: b.USES_DEFAULT_IMPLEMENTATION
          }
      }
  });
  e.default = A
}
