// react-native-maps/lib/components/MapPolyline.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      o = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[2])),
      l = t(r(d[3])),
      u = t(r(d[4])),
      f = t(r(d[5])),
      s = t(r(d[6])),
      p = t(r(d[7])),
      c = t(r(d[8])),
      y = t(r(d[9])),
      b = t(r(d[10])),
      P = r(d[11]),
      O = o(r(d[12]));

  function v(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }
  var h = (function(t) {
          for (var o = 1; o < arguments.length; o++) {
              var n = null != arguments[o] ? arguments[o] : {};
              o % 2 ? v(n, !0).forEach(function(o) {
                  (0, c.default)(t, o, n[o])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(n)) : v(n).forEach(function(o) {
                  Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(n, o))
              })
          }
          return t
      })({}, P.ViewPropTypes || P.View.propTypes, {
          coordinates: y.default.arrayOf(y.default.shape({
              latitude: y.default.number.isRequired,
              longitude: y.default.number.isRequired
          })),
          onPress: y.default.func,
          tappable: y.default.bool,
          fillColor: P.ColorPropType,
          strokeWidth: y.default.number,
          strokeColor: P.ColorPropType,
          strokeColors: y.default.arrayOf(P.ColorPropType),
          zIndex: y.default.number,
          lineCap: y.default.oneOf(['butt', 'round', 'square']),
          lineJoin: y.default.oneOf(['miter', 'round', 'bevel']),
          miterLimit: y.default.number,
          geodesic: y.default.bool,
          lineDashPhase: y.default.number,
          lineDashPattern: y.default.arrayOf(y.default.number)
      }),
      j = (function(t) {
          function o() {
              return (0, l.default)(this, o), (0, f.default)(this, (0, s.default)(o).apply(this, arguments))
          }
          return (0, p.default)(o, t), (0, u.default)(o, [{
              key: "setNativeProps",
              value: function(t) {
                  this.polyline.setNativeProps(t)
              }
          }, {
              key: "render",
              value: function() {
                  var t = this,
                      o = this.getAirComponent();
                  return b.default.createElement(o, (0, n.default)({}, this.props, {
                      ref: function(o) {
                          t.polyline = o
                      },
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapPolyline.js",
                          lineNumber: 148
                      }
                  }))
              }
          }]), o
      })(b.default.Component);
  j.propTypes = h, j.defaultProps = {
      strokeColor: '#000',
      strokeWidth: 1,
      lineJoin: 'round',
      lineCap: 'round'
  };
  var C = (0, O.default)(j, {
      componentType: 'Polyline',
      providers: {
          google: {
              ios: O.SUPPORTED,
              android: O.USES_DEFAULT_IMPLEMENTATION
          }
      }
  });
  e.default = C
}
