// react-native-maps/lib/components/MapUrlTile.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[2])),
      u = t(r(d[3])),
      l = t(r(d[4])),
      p = t(r(d[5])),
      f = t(r(d[6])),
      s = t(r(d[7])),
      c = t(r(d[8])),
      b = t(r(d[9])),
      y = t(r(d[10])),
      O = r(d[11]),
      v = n(r(d[12]));

  function P(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var u = Object.getOwnPropertySymbols(t);
          n && (u = u.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, u)
      }
      return o
  }
  var j = (function(t) {
          for (var n = 1; n < arguments.length; n++) {
              var o = null != arguments[n] ? arguments[n] : {};
              n % 2 ? P(o, !0).forEach(function(n) {
                  (0, c.default)(t, n, o[n])
              }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : P(o).forEach(function(n) {
                  Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
              })
          }
          return t
      })({}, O.ViewPropTypes || O.View.propTypes, {
          urlTemplate: b.default.string.isRequired,
          zIndex: b.default.number,
          maximumZ: b.default.number,
          minimumZ: b.default.number,
          shouldReplaceMapContent: b.default.bool,
          tileSize: b.default.number,
          flipY: b.default.bool
      }),
      T = (function(t) {
          function n() {
              return (0, u.default)(this, n), (0, p.default)(this, (0, f.default)(n).apply(this, arguments))
          }
          return (0, s.default)(n, t), (0, l.default)(n, [{
              key: "render",
              value: function() {
                  var t = this.getAirComponent();
                  return y.default.createElement(t, (0, o.default)({}, this.props, {
                      __source: {
                          fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-maps/lib/components/MapUrlTile.js",
                          lineNumber: 69
                      }
                  }))
              }
          }]), n
      })(y.default.Component);
  T.propTypes = j;
  var h = (0, v.default)(T, {
      componentType: 'UrlTile',
      providers: {
          google: {
              ios: v.SUPPORTED,
              android: v.USES_DEFAULT_IMPLEMENTATION
          }
      }
  });
  e.default = h
}
