// react-native-flash-message/src/FlashMessage.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.positionStyle = k, e.showMessage = function() {
      var t = S.default.getDefault();
      t && t.showMessage.apply(t, arguments)
  }, e.hideMessage = function() {
      var t = S.default.getDefault();
      t && t.hideMessage.apply(t, arguments)
  }, e.FlashMessageTransition = H, e.default = e.DefaultFlash = e.renderFlashMessageIcon = void 0;
  var o = t(r(d[2])),
      s = t(r(d[3])),
      l = t(r(d[4])),
      u = t(r(d[5])),
      f = t(r(d[6])),
      c = t(r(d[7])),
      h = t(r(d[8])),
      p = t(r(d[9])),
      y = t(r(d[10])),
      b = n(r(d[11])),
      v = r(d[12]),
      C = r(d[13]),
      M = t(r(d[14])),
      S = t(r(d[15])),
      _ = n(r(d[16])),
      T = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-flash-message/src/FlashMessage.js";

  function P(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }

  function I(t) {
      for (var n = 1; n < arguments.length; n++) {
          var o = null != arguments[n] ? arguments[n] : {};
          n % 2 ? P(o, !0).forEach(function(n) {
              (0, y.default)(t, n, o[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : P(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var N = "ios" !== v.Platform.OS ? 60 : 48,
      w = M.default.shape({
          message: M.default.string.isRequired,
          description: M.default.string,
          type: M.default.string,
          backgroundColor: M.default.string,
          color: M.default.string
      }).isRequired,
      F = function() {};

  function O() {
      function t() {
          return Math.floor(65536 * (1 + Math.random())).toString(16).substring(1)
      }
      return t() + "-" + t() + "-" + t()
  }

  function D() {
      var t = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "none";
      return t && "none" !== t ? "string" == typeof t ? {
          icon: t,
          position: "left",
          style: {}
      } : I({
          position: "left",
          style: {}
      }, t) : null
  }

  function k(t, n) {
      return "string" == typeof n ? [t, "top" === n && j.rootTop, "bottom" === n && j.rootBottom, "center" === n && j.rootCenter] : [t, n]
  }

  function H(t) {
      var n = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : "top",
          o = t.interpolate({
              inputRange: [0, 1],
              outputRange: [0, 1]
          });
      return "top" === n ? {
          transform: [{
              translateY: t.interpolate({
                  inputRange: [0, 1],
                  outputRange: [-N, 0]
              })
          }],
          opacity: o
      } : "bottom" === n ? {
          transform: [{
              translateY: t.interpolate({
                  inputRange: [0, 1],
                  outputRange: [N, 0]
              })
          }],
          opacity: o
      } : {
          opacity: o
      }
  }
  var R = function() {
      var t = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : "success",
          n = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : {},
          o = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : {};
      switch (t) {
          case "success":
              return b.default.createElement(v.Image, (0, p.default)({
                  style: [j.flashIcon, n],
                  source: r(d[17])
              }, o, {
                  __source: {
                      fileName: T,
                      lineNumber: 148
                  }
              }));
          case "info":
              return b.default.createElement(v.Image, (0, p.default)({
                  style: [j.flashIcon, n],
                  source: r(d[18])
              }, o, {
                  __source: {
                      fileName: T,
                      lineNumber: 151
                  }
              }));
          case "warning":
              return b.default.createElement(v.Image, (0, p.default)({
                  style: [j.flashIcon, n],
                  source: r(d[19])
              }, o, {
                  __source: {
                      fileName: T,
                      lineNumber: 154
                  }
              }));
          case "danger":
              return b.default.createElement(v.Image, (0, p.default)({
                  style: [j.flashIcon, n],
                  source: r(d[20])
              }, o, {
                  __source: {
                      fileName: T,
                      lineNumber: 158
                  }
              }));
          default:
              return null
      }
  };
  e.renderFlashMessageIcon = R;
  var V = function(t) {
      var n = t.message,
          o = t.style,
          s = t.textStyle,
          l = t.titleStyle,
          u = t.renderFlashMessageIcon,
          f = t.position,
          c = void 0 === f ? "top" : f,
          y = t.renderCustomContent,
          C = t.floating,
          M = void 0 !== C && C,
          S = t.icon,
          P = t.hideStatusBar,
          I = void 0 !== P && P,
          N = (0, h.default)(t, ["message", "style", "textStyle", "titleStyle", "renderFlashMessageIcon", "position", "renderCustomContent", "floating", "icon", "hideStatusBar"]),
          w = !!n.description && "" !== n.description,
          F = !!S && !!S.icon && u("auto" === S.icon ? n.type : S.icon, ["left" === S.position && j.flashIconLeft, "right" === S.position && j.flashIconRight, S.style]),
          O = !!F;
      return b.default.createElement(_.default, {
          position: "string" == typeof c ? c : null,
          __source: {
              fileName: T,
              lineNumber: 195
          }
      }, function(t) {
          return b.default.createElement(v.View, (0, p.default)({
              style: (0, _.styleWithInset)([j.defaultFlash, "center" === c && j.defaultFlashCenter, "center" !== c && M && j.defaultFlashFloating, O && j.defaultFlashWithIcon, n.backgroundColor ? {
                  backgroundColor: n.backgroundColor
              } : !!n.type && !!E.ColorTheme[n.type] && {
                  backgroundColor: E.ColorTheme[n.type]
              }, o], t, !!I, "center" !== c && M ? "margin" : "padding")
          }, N, {
              __source: {
                  fileName: T,
                  lineNumber: 197
              }
          }), O && "left" === S.position && F, b.default.createElement(v.View, {
              style: j.flashLabel,
              __source: {
                  fileName: T,
                  lineNumber: 218
              }
          }, b.default.createElement(v.Text, {
              style: [j.flashText, w && j.flashTitle, !!n.color && {
                  color: n.color
              }, l],
              __source: {
                  fileName: T,
                  lineNumber: 219
              }
          }, n.message), !!y && y(), w && b.default.createElement(v.Text, {
              style: [j.flashText, !!n.color && {
                  color: n.color
              }, s],
              __source: {
                  fileName: T,
                  lineNumber: 230
              }
          }, n.description)), O && "right" === S.position && F)
      })
  };
  e.DefaultFlash = V, V.propTypes = {
      message: w,
      renderFlashMessageIcon: M.default.func
  };
  var E = (function(t) {
      function n(t) {
          var s;
          return (0, o.default)(this, n), (s = (0, l.default)(this, (0, u.default)(n).call(this, t))).prop = s.prop.bind((0, f.default)(s)), s.pressMessage = s.pressMessage.bind((0, f.default)(s)), s.longPressMessage = s.longPressMessage.bind((0, f.default)(s)), s.toggleVisibility = s.toggleVisibility.bind((0, f.default)(s)), s._id || (s._id = O()), s.state = {
              visibleValue: new v.Animated.Value(0),
              isHidding: !1,
              message: t.message || null
          }, s
      }
      return (0, c.default)(n, t), (0, s.default)(n, [{
          key: "componentDidMount",
          value: function() {
              this.props.canRegisterAsDefault && S.default.register(this)
          }
      }, {
          key: "componentWillUnmount",
          value: function() {
              this.props.canRegisterAsDefault && S.default.unregister(this)
          }
      }, {
          key: "prop",
          value: function(t, n) {
              return t && n in t ? t[n] : n in this.props ? this.props[n] : null
          }
      }, {
          key: "isAnimated",
          value: function(t) {
              return this.prop(t, "animated")
          }
      }, {
          key: "pressMessage",
          value: function(t) {
              if (!this.state.isHidding) {
                  var n = this.state.message,
                      o = this.prop(n, "hideOnPress"),
                      s = this.prop(n, "onPress");
                  o && this.hideMessage(), "function" == typeof s && s(t, n)
              }
          }
      }, {
          key: "longPressMessage",
          value: function(t) {
              if (!this.state.isHidding) {
                  var n = this.state.message,
                      o = this.prop(n, "hideOnPress"),
                      s = this.prop(n, "onLongPress");
                  o && this.hideMessage(), "function" == typeof s && s(t, n)
              }
          }
      }, {
          key: "toggleVisibility",
          value: function() {
              var t = this,
                  n = !(arguments.length > 0 && void 0 !== arguments[0]) || arguments[0],
                  o = !(arguments.length > 1 && void 0 !== arguments[1]) || arguments[1],
                  s = arguments.length > 2 ? arguments[2] : void 0,
                  l = this.state.message,
                  u = (this.prop(l, "position"), this.prop(l, "animationDuration")),
                  f = this.prop(l, "duration"),
                  c = this.prop(l, "autoHide"),
                  h = this.prop(l, "hideStatusBar");
              if (this._hideTimeout && clearTimeout(this._hideTimeout), n) {
                  var p = this.prop(l, "onShow") || F,
                      y = function() {
                          c && f > 0 && (t._hideTimeout = setTimeout(function() {
                              return t.toggleVisibility(!1, o)
                          }, f)), s && "function" == typeof s && s()
                      };
                  this.setState({
                      isHidding: !1
                  }), this.state.visibleValue.setValue(0), p && "function" == typeof p && p(this), h && v.StatusBar.setHidden(!0, "string" == typeof h ? h : "slide"), o ? v.Animated.timing(this.state.visibleValue, {
                      toValue: 1,
                      duration: u,
                      useNativeDriver: !0
                  }).start(y) : y()
              } else {
                  var b = this.prop(l, "onHide") || F,
                      C = function() {
                          t.setState({
                              message: null,
                              isHidding: !1
                          }), b && "function" == typeof b && b(t), s && "function" == typeof s && s()
                      };
                  this.setState({
                      isHidding: !0
                  }), h && v.StatusBar.setHidden(!1, "string" == typeof h ? h : "slide"), o ? v.Animated.timing(this.state.visibleValue, {
                      toValue: 0,
                      duration: u,
                      useNativeDriver: !0
                  }).start(C) : C()
              }
          }
      }, {
          key: "showMessage",
          value: function(t) {
              var n = this,
                  o = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : null,
                  s = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : "default";
              if (t) {
                  var l = {};
                  "string" == typeof t ? l = {
                      message: t,
                      description: o,
                      type: s
                  } : "message" in t && (l = I({
                      description: null,
                      type: "default"
                  }, t));
                  var u = this.isAnimated(l);
                  this.setState({
                      message: l
                  }, function() {
                      return n.toggleVisibility(!0, u)
                  })
              } else this.setState({
                  message: null,
                  isHidding: !1
              })
          }
      }, {
          key: "hideMessage",
          value: function() {
              var t = this.isAnimated(this.state.message);
              this.toggleVisibility(!1, t)
          }
      }, {
          key: "render",
          value: function() {
              var t = this.props,
                  n = t.renderFlashMessageIcon,
                  o = t.renderCustomContent,
                  s = t.MessageComponent,
                  l = this.state,
                  u = l.message,
                  f = l.visibleValue,
                  c = this.prop(u, "style"),
                  h = this.prop(u, "textStyle"),
                  p = this.prop(u, "titleStyle"),
                  y = this.prop(u, "floating"),
                  C = this.prop(u, "position"),
                  M = D(this.prop(u, "icon")),
                  S = this.prop(u, "hideStatusBar"),
                  _ = this.prop(u, "transitionConfig"),
                  P = this.isAnimated(u) ? _(f, C) : {};
              return b.default.createElement(v.Animated.View, {
                  style: [k(j.root, C), "center" === C && !!u && j.rootCenterEnabled, P],
                  __source: {
                      fileName: T,
                      lineNumber: 567
                  }
              }, !!u && b.default.createElement(v.TouchableWithoutFeedback, {
                  onPress: this.pressMessage,
                  onLongPress: this.longPressMessage,
                  __source: {
                      fileName: T,
                      lineNumber: 574
                  }
              }, b.default.createElement(s, {
                  position: C,
                  floating: y,
                  message: u,
                  hideStatusBar: S,
                  renderFlashMessageIcon: n,
                  renderCustomContent: o,
                  icon: M,
                  style: c,
                  textStyle: h,
                  titleStyle: p,
                  __source: {
                      fileName: T,
                      lineNumber: 575
                  }
              })))
          }
      }]), n
  })(b.Component);
  e.default = E, E.defaultProps = {
      canRegisterAsDefault: !0,
      hideOnPress: !0,
      onPress: F,
      onLongPress: F,
      animated: !0,
      animationDuration: 225,
      autoHide: !0,
      duration: 1850,
      hideStatusBar: !1,
      floating: !1,
      position: "top",
      renderCustomContent: null,
      icon: "none",
      renderFlashMessageIcon: R,
      transitionConfig: H,
      MessageComponent: V
  }, E.propTypes = {
      canRegisterAsDefault: M.default.bool,
      hideOnPress: M.default.bool,
      onShow: M.default.func,
      onHide: M.default.func,
      onPress: M.default.func,
      onLongPress: M.default.func,
      animated: M.default.bool,
      animationDuration: M.default.number,
      duration: M.default.number,
      autoHide: M.default.bool,
      hideStatusBar: M.default.bool,
      floating: M.default.bool,
      position: M.default.oneOfType([M.default.string, M.default.array, M.default.object]),
      renderCustomContent: M.default.func,
      icon: M.default.oneOfType([M.default.string, M.default.object]),
      renderFlashMessageIcon: M.default.func,
      transitionConfig: M.default.func,
      MessageComponent: M.default.oneOfType([M.default.element, M.default.func])
  }, E.ColorTheme = {
      success: "#5cb85c",
      info: "#5bc0de",
      warning: "#f0ad4e",
      danger: "#d9534f"
  }, E.setColorTheme = function(t) {
      E.ColorTheme = (0, p.default)(E.ColorTheme, t)
  };
  var j = v.StyleSheet.create({
      root: {
          position: "absolute",
          left: 0,
          right: 0
      },
      rootTop: {
          top: 0
      },
      rootBottom: {
          bottom: 0
      },
      rootCenter: {
          justifyContent: "center",
          alignItems: "center"
      },
      rootCenterEnabled: {
          top: 0,
          bottom: 0,
          width: "100%",
          height: "100%"
      },
      defaultFlash: {
          justifyContent: "flex-start",
          paddingVertical: 15,
          paddingHorizontal: 20,
          backgroundColor: "#696969",
          minHeight: N
      },
      defaultFlashCenter: {
          margin: 44,
          borderRadius: 8,
          overflow: "hidden"
      },
      defaultFlashFloating: {
          marginTop: 10,
          marginLeft: 12,
          marginRight: 12,
          marginBottom: 10,
          borderRadius: 8,
          minHeight: N - (0, C.getStatusBarHeight)()
      },
      defaultFlashWithIcon: {
          flexDirection: "row"
      },
      flashLabel: {
          flexDirection: "column"
      },
      flashText: {
          fontSize: 14,
          lineHeight: 18,
          color: "#fff"
      },
      flashTitle: {
          fontSize: 16,
          fontWeight: "600",
          marginBottom: 5
      },
      flashIcon: {
          tintColor: "#fff",
          marginTop: -1,
          width: 21,
          height: 21
      },
      flashIconLeft: {
          marginLeft: -6,
          marginRight: 9
      },
      flashIconRight: {
          marginRight: -6,
          marginLeft: 9
      }
  })
}
