// @unimodules/react-native-adapter/build/NativeViewManagerAdapter.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.requireNativeViewManager = function(t) {
      var v = "ViewManagerAdapter_" + t,
          w = (0, s.requireNativeComponent)(v),
          y = (s.UIManager.getViewManagerConfig ? s.UIManager.getViewManagerConfig(v) : s.UIManager[v]) || {
              NativeProps: {},
              directEventTypes: {}
          },
          M = ['children'].concat((0, o.default)(c), (0, o.default)(Object.keys(y.NativeProps)), (0, o.default)(Object.keys(y.directEventTypes)));

      function N(t, o) {
          var s = (0, l.default)(t, M),
              c = (0, u.default)(t, M);
          return f.default.createElement(w, (0, n.default)({}, s, {
              proxiedProperties: c,
              ref: o,
              __source: {
                  fileName: p,
                  lineNumber: 49
              }
          }))
      }
      return N.displayName = "Adapter<" + t + ">", f.default.forwardRef(N)
  };
  var n = t(r(d[1])),
      o = t(r(d[2])),
      u = t(r(d[3])),
      l = t(r(d[4])),
      f = t(r(d[5])),
      s = r(d[6]),
      p = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/@unimodules/react-native-adapter/build/NativeViewManagerAdapter.js",
      c = Object.keys(s.ViewPropTypes)
}
