// react-native/Libraries/Components/Button.js
module.exports = function(g, r, i, a, m, e, d) {
  'use strict';
  var t = r(d[0]),
      o = r(d[1]),
      n = r(d[2]),
      s = r(d[3]),
      c = r(d[4]),
      l = r(d[5]),
      u = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native/Libraries/Components/Button.js";

  function b(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          o && (s = s.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, s)
      }
      return n
  }
  r(d[6]);
  var p = r(d[7]),
      f = r(d[8]),
      y = r(d[9]),
      h = r(d[10]),
      O = (r(d[11]), r(d[12])),
      v = r(d[13]),
      P = (function(t) {
          function b() {
              return o(this, b), s(this, c(b).apply(this, arguments))
          }
          return l(b, t), n(b, [{
              key: "render",
              value: function() {
                  var t = this.props,
                      o = t.accessibilityLabel,
                      n = t.color,
                      s = t.onPress,
                      c = t.title,
                      l = t.hasTVPreferredFocus,
                      b = t.disabled,
                      f = t.testID,
                      P = [D.button],
                      j = [D.text];
                  n && P.push({
                      backgroundColor: n
                  });
                  var w = [];
                  b && (P.push(D.buttonDisabled), j.push(D.textDisabled), w.push('disabled')), v('string' == typeof c, 'The title prop of a Button must be a string');
                  var _ = c.toUpperCase(),
                      k = h;
                  return p.createElement(k, {
                      accessibilityLabel: o,
                      accessibilityRole: "button",
                      accessibilityStates: w,
                      hasTVPreferredFocus: l,
                      testID: f,
                      disabled: b,
                      onPress: s,
                      __source: {
                          fileName: u,
                          lineNumber: 125
                      }
                  }, p.createElement(O, {
                      style: P,
                      __source: {
                          fileName: u,
                          lineNumber: 133
                      }
                  }, p.createElement(y, {
                      style: j,
                      disabled: b,
                      __source: {
                          fileName: u,
                          lineNumber: 134
                      }
                  }, _)))
              }
          }]), b
      })(p.Component),
      D = f.create({
          button: {
              elevation: 4,
              backgroundColor: '#2196F3',
              borderRadius: 2
          },
          text: (function(o) {
              for (var n = 1; n < arguments.length; n++) {
                  var s = null != arguments[n] ? arguments[n] : {};
                  n % 2 ? b(s, !0).forEach(function(n) {
                      t(o, n, s[n])
                  }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(o, Object.getOwnPropertyDescriptors(s)) : b(s).forEach(function(t) {
                      Object.defineProperty(o, t, Object.getOwnPropertyDescriptor(s, t))
                  })
              }
              return o
          })({
              textAlign: 'center',
              padding: 8
          }, {
              color: 'white',
              fontWeight: '500'
          }),
          buttonDisabled: {
              elevation: 0,
              backgroundColor: '#dfdfdf'
          },
          textDisabled: {
              color: '#a1a1a1'
          }
      });
  m.exports = P
}
