// react-native-safe-area-view/index.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]),
      n = r(d[1]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.getInset = E, e.getStatusBarHeight = function(t) {
      return N(t)
  }, e.withSafeArea = e.default = void 0;
  var o = t(r(d[2])),
      s = t(r(d[3])),
      u = t(r(d[4])),
      p = t(r(d[5])),
      f = t(r(d[6])),
      c = t(r(d[7])),
      l = t(r(d[8])),
      h = t(r(d[9])),
      v = n(r(d[10])),
      y = r(d[11]),
      w = t(r(d[12])),
      b = t(r(d[13])),
      _ = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-safe-area-view/index.js";

  function S(t, n) {
      var o = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var s = Object.getOwnPropertySymbols(t);
          n && (s = s.filter(function(n) {
              return Object.getOwnPropertyDescriptor(t, n).enumerable
          })), o.push.apply(o, s)
      }
      return o
  }

  function L(t) {
      for (var n = 1; n < arguments.length; n++) {
          var o = null != arguments[n] ? arguments[n] : {};
          n % 2 ? S(o, !0).forEach(function(n) {
              (0, u.default)(t, n, o[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(o)) : S(o).forEach(function(n) {
              Object.defineProperty(t, n, Object.getOwnPropertyDescriptor(o, n))
          })
      }
      return t
  }
  var O = function() {
          var t = y.Dimensions.get('window'),
              n = t.width,
              o = t.height;
          return 0 === n && 0 === o ? y.Dimensions.get('screen') : {
              width: n,
              height: o
          }
      },
      B = O(),
      P = B.height,
      I = B.width,
      M = (((y.Platform.constants || {}).reactNativeVersion || {}).minor, 'web' !== y.Platform.OS && ('ios' === y.Platform.OS && (812 === P && 375 === I || 375 === P && 812 === I) || 896 === P && 414 === I || 414 === P && 896 === I)),
      j = 'ios' === y.Platform.OS && (1194 === P && 834 === I || 834 === P && 1194 === I || 1024 === P && 1366 === I || 1366 === P && 1024 === I),
      R = !('ios' !== y.Platform.OS || M || P > I && I < 768 || I > P && P < 768),
      T = null,
      k = null,
      N = function(t) {
          return null !== T ? T : 'android' === y.Platform.OS ? g.Expo ? g.Expo.Constants.statusBarHeight : 0 : M ? t ? 0 : 44 : j ? 24 : R ? k ? 0 : 20 : t || k ? 0 : 20
      },
      x = function(t) {
          if (!t.includes('%')) return 0;
          var n = parseFloat(t) / 100;
          return isNaN(n) ? 0 : n
      },
      D = (function(t) {
          function n() {
              var t, o;
              (0, p.default)(this, n);
              for (var u = arguments.length, f = new Array(u), h = 0; h < u; h++) f[h] = arguments[h];
              return (o = (0, c.default)(this, (t = (0, l.default)(n)).call.apply(t, [this].concat(f)))).state = {
                  touchesTop: !0,
                  touchesBottom: !0,
                  touchesLeft: !0,
                  touchesRight: !0,
                  orientation: null,
                  viewWidth: 0,
                  viewHeight: 0
              }, o._handleLayout = function(t) {
                  o.props.onLayout && o.props.onLayout(t), o._updateMeasurements()
              }, o._updateMeasurements = function() {
                  if (o._isMounted && o.view) {
                      var t = o.props.isLandscape,
                          n = o.state.orientation,
                          s = t ? 'landscape' : 'portrait';
                      if (!n || n !== s) {
                          var u = O(),
                              p = u.width,
                              f = u.height;
                          o.view.getNode().measureInWindow(function(t, n, u, c) {
                              if (o.view) {
                                  var l = n,
                                      h = t;
                                  l >= f ? l %= f : l < 0 && (l = l % f + f), h >= p ? h %= p : h < 0 && (h = h % p + p);
                                  var v = 0 === l,
                                      y = l + c >= f,
                                      w = 0 === h,
                                      b = h + u >= p;
                                  o.setState({
                                      touchesTop: v,
                                      touchesBottom: y,
                                      touchesLeft: w,
                                      touchesRight: b,
                                      orientation: s,
                                      viewWidth: u,
                                      viewHeight: c
                                  })
                              }
                          })
                      }
                  }
              }, o._getSafeAreaStyle = function() {
                  var t = o.state,
                      n = t.touchesTop,
                      s = t.touchesBottom,
                      u = t.touchesLeft,
                      p = t.touchesRight,
                      f = o.props,
                      c = f.forceInset,
                      l = (f.isLandscape, o._getViewStyles()),
                      h = l.paddingTop,
                      v = l.paddingBottom,
                      y = l.paddingLeft,
                      w = l.paddingRight,
                      b = L({}, l.viewStyle, {
                          paddingTop: n ? o._getInset('top') : 0,
                          paddingBottom: s ? o._getInset('bottom') : 0,
                          paddingLeft: u ? o._getInset('left') : 0,
                          paddingRight: p ? o._getInset('right') : 0
                      });
                  return c && Object.keys(c).forEach(function(t) {
                      var n = c[t];
                      switch ('always' === n && (n = o._getInset(t)), 'never' === n && (n = 0), t) {
                          case 'horizontal':
                              b.paddingLeft = n, b.paddingRight = n;
                              break;
                          case 'vertical':
                              b.paddingTop = n, b.paddingBottom = n;
                              break;
                          case 'left':
                          case 'right':
                          case 'top':
                          case 'bottom':
                              var s = "padding" + t[0].toUpperCase() + t.slice(1);
                              b[s] = n
                      }
                  }), b.height && 'number' == typeof b.height && (b.height += b.paddingTop + b.paddingBottom), b.width && 'number' == typeof b.width && (b.width += b.paddingLeft + b.paddingRight), b.paddingTop = Math.max(b.paddingTop, h), b.paddingBottom = Math.max(b.paddingBottom, v), b.paddingLeft = Math.max(b.paddingLeft, y), b.paddingRight = Math.max(b.paddingRight, w), b
              }, o._getViewStyles = function() {
                  var t = o.state.viewWidth,
                      n = y.StyleSheet.flatten(o.props.style || {}),
                      u = n.padding,
                      p = void 0 === u ? 0 : u,
                      f = n.paddingVertical,
                      c = void 0 === f ? p : f,
                      l = n.paddingHorizontal,
                      h = void 0 === l ? p : l,
                      v = n.paddingTop,
                      w = void 0 === v ? c : v,
                      b = n.paddingBottom,
                      _ = void 0 === b ? c : b,
                      S = n.paddingLeft,
                      L = void 0 === S ? h : S,
                      O = n.paddingRight,
                      B = void 0 === O ? h : O,
                      P = (0, s.default)(n, ["padding", "paddingVertical", "paddingHorizontal", "paddingTop", "paddingBottom", "paddingLeft", "paddingRight"]);
                  return 'number' != typeof w && (w = x(w) * t), 'number' != typeof _ && (_ = x(_) * t), 'number' != typeof L && (L = x(L) * t), 'number' != typeof B && (B = x(B) * t), {
                      paddingTop: w,
                      paddingBottom: _,
                      paddingLeft: L,
                      paddingRight: B,
                      viewStyle: P
                  }
              }, o._getInset = function(t) {
                  return E(t, o.props.isLandscape)
              }, o
          }
          return (0, h.default)(n, t), (0, f.default)(n, [{
              key: "componentDidMount",
              value: function() {
                  var t = this;
                  this._isMounted = !0, y.InteractionManager.runAfterInteractions(function() {
                      t._updateMeasurements()
                  })
              }
          }, {
              key: "componentWillUnmount",
              value: function() {
                  this._isMounted = !1
              }
          }, {
              key: "componentDidUpdate",
              value: function() {
                  this._updateMeasurements()
              }
          }, {
              key: "render",
              value: function() {
                  var t = this,
                      n = this.props,
                      u = (n.forceInset, n.isLandscape, n.style, (0, s.default)(n, ["forceInset", "isLandscape", "style"])),
                      p = this._getSafeAreaStyle();
                  return v.default.createElement(y.Animated.View, (0, o.default)({
                      ref: function(n) {
                          return t.view = n
                      },
                      pointerEvents: "box-none"
                  }, u, {
                      onLayout: this._handleLayout,
                      style: p,
                      __source: {
                          fileName: _,
                          lineNumber: 163
                      }
                  }))
              }
          }]), n
      })(v.Component);

  function E(t, n) {
      switch (t) {
          case 'horizontal':
          case 'right':
          case 'left':
              return n && M ? 44 : 0;
          case 'vertical':
          case 'top':
              return N(n);
          case 'bottom':
              return M ? n ? 24 : 34 : j ? 20 : 0
      }
  }
  D.setStatusBarHeight = function(t) {
      T = t
  }, D.setStatusBarHidden = function(t) {
      k = t
  };
  var A = (0, b.default)(D),
      H = A;
  e.default = H;
  e.withSafeArea = function() {
      var t = arguments.length > 0 && void 0 !== arguments[0] ? arguments[0] : {};
      return function(n) {
          var s = (function(s) {
              function u() {
                  return (0, p.default)(this, u), (0, c.default)(this, (0, l.default)(u).apply(this, arguments))
              }
              return (0, h.default)(u, s), (0, f.default)(u, [{
                  key: "render",
                  value: function() {
                      return v.default.createElement(A, {
                          style: {
                              flex: 1
                          },
                          forceInset: t,
                          __source: {
                              fileName: _,
                              lineNumber: 385
                          }
                      }, v.default.createElement(n, (0, o.default)({}, this.props, {
                          __source: {
                              fileName: _,
                              lineNumber: 386
                          }
                      })))
                  }
              }]), u
          })(v.Component);
          return (0, w.default)(s, n)
      }
  }
}
