// react-native-onboarding-swiper/src/buttons/DoneButton.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var n = t(r(d[1])),
      u = t(r(d[2])),
      l = t(r(d[3])),
      o = t(r(d[4])),
      s = t(r(d[5])),
      f = t(r(d[6])),
      c = t(r(d[7])),
      _ = r(d[8]),
      v = t(r(d[9])),
      y = r(d[10]),
      p = t(r(d[11])),
      h = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-onboarding-swiper/src/buttons/DoneButton.js",
      A = (function(t) {
          function A() {
              var t, n;
              (0, l.default)(this, A);
              for (var u = arguments.length, o = new Array(u), c = 0; c < u; c++) o[c] = arguments[c];
              return (n = (0, s.default)(this, (t = (0, f.default)(A)).call.apply(t, [this].concat(o)))).state = {
                  fadeAnim: new _.Animated.Value(0)
              }, n
          }
          return (0, c.default)(A, t), (0, o.default)(A, [{
              key: "componentDidMount",
              value: function() {
                  var t = this;
                  setTimeout(function() {
                      _.Animated.timing(t.state.fadeAnim, {
                          toValue: 1,
                          duration: 1e3
                      }).start()
                  }, 1e3)
              }
          }, {
              key: "render",
              value: function() {
                  var t = this.props,
                      l = t.isLight,
                      o = (0, u.default)(t, ["isLight"]),
                      s = this.state.fadeAnim;
                  return v.default.createElement(_.Animated.View, {
                      style: {
                          opacity: s
                      },
                      __source: {
                          fileName: h,
                          lineNumber: 26
                      }
                  }, v.default.createElement(p.default, (0, n.default)({
                      size: y.BUTTON_SIZE,
                      textStyle: (0, y.getDefaultStyle)(l),
                      style: {
                          borderRadius: y.BUTTON_SIZE / 2,
                          backgroundColor: 'rgba(255, 255, 255, 0.10)',
                          margin: y.MARGIN_RIGHT
                      }
                  }, o, {
                      __source: {
                          fileName: h,
                          lineNumber: 31
                      }
                  }), "\u2713"))
              }
          }]), A
      })(v.default.Component);
  e.default = A
}
