// react-native-onboarding-swiper/src/buttons/TextButton.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = void 0;
  var o = t(r(d[1])),
      n = r(d[2]),
      l = t(r(d[3])),
      s = t(r(d[4])),
      u = "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-onboarding-swiper/src/buttons/TextButton.js";

  function c(t, o) {
      var n = Object.keys(t);
      if (Object.getOwnPropertySymbols) {
          var l = Object.getOwnPropertySymbols(t);
          o && (l = l.filter(function(o) {
              return Object.getOwnPropertyDescriptor(t, o).enumerable
          })), n.push.apply(n, l)
      }
      return n
  }

  function f(t) {
      for (var n = 1; n < arguments.length; n++) {
          var l = null != arguments[n] ? arguments[n] : {};
          n % 2 ? c(l, !0).forEach(function(n) {
              (0, o.default)(t, n, l[n])
          }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(t, Object.getOwnPropertyDescriptors(l)) : c(l).forEach(function(o) {
              Object.defineProperty(t, o, Object.getOwnPropertyDescriptor(l, o))
          })
      }
      return t
  }
  var p = function(t) {
      var o = t.size,
          s = t.onPress,
          c = t.textStyle,
          p = t.allowFontScaling,
          y = t.style,
          b = t.children;
      return l.default.createElement(n.View, {
          style: f({
              flex: 0,
              paddingHorizontal: 10
          }, y),
          __source: {
              fileName: u,
              lineNumber: 6
          }
      }, l.default.createElement(n.TouchableOpacity, {
          style: {
              flex: 0
          },
          onPress: s,
          hitSlop: {
              top: 15,
              bottom: 15,
              left: 15,
              right: 15
          },
          __source: {
              fileName: u,
              lineNumber: 7
          }
      }, l.default.createElement(n.Text, {
          allowFontScaling: p,
          style: f({
              fontSize: o / 2.5
          }, c),
          __source: {
              fileName: u,
              lineNumber: 12
          }
      }, b)))
  };
  p.propTypes = {
      size: s.default.number.isRequired,
      onPress: s.default.func.isRequired,
      textStyle: n.Text.propTypes.style,
      allowFontScaling: s.default.bool
  }, p.defaultProps = {
      textStyle: null,
      allowFontScaling: !0
  };
  var y = p;
  e.default = y
}
