// react-native-elements/src/divider/Divider.js
module.exports = function(g, r, i, a, m, e, d) {
  var t = r(d[0]);
  Object.defineProperty(e, "__esModule", {
      value: !0
  }), e.default = e.Divider = void 0;
  var l = t(r(d[1])),
      n = t(r(d[2])),
      o = t(r(d[3])),
      s = t(r(d[4])),
      u = r(d[5]),
      c = r(d[6]),
      v = function(t) {
          var s = t.style,
              c = t.theme,
              v = (0, n.default)(t, ["style", "theme"]);
          return o.default.createElement(u.View, (0, l.default)({
              style: u.StyleSheet.flatten([f.container(c), s])
          }, v, {
              __source: {
                  fileName: "/Users/linyiting/Desktop/Arrow/arrow/node_modules/react-native-elements/src/divider/Divider.js",
                  lineNumber: 8
              }
          }))
      };
  e.Divider = v, v.propTypes = {
      style: c.ViewPropTypes.style,
      theme: s.default.object
  };
  var f = {
          container: function(t) {
              return {
                  backgroundColor: t.colors.divider,
                  height: u.StyleSheet.hairlineWidth
              }
          }
      },
      h = (0, c.withTheme)(v, 'Divider');
  e.default = h
}
